ccl/backupccl: TestBackupRestoreWithConcurrentWrites failed

[(ccl/backupccl).TestBackupRestoreWithConcurrentWrites failed](https://viewLog.html?buildId=1648687&tab=buildLog) on [master@20e2f036ac18eed4ee2d58958bb8e725add9f4fa](https://github.com/cockroachdb/cockroach/commits/20e2f036ac18eed4ee2d58958bb8e725add9f4fa):

```
E191217 17:31:04.478309 171977 sql/lease.go:1194  background lease renewal for tableID=53 tableName="bank" failed: table is being dropped
E191217 17:31:04.601871 171982 sql/lease.go:1194  background lease renewal for tableID=53 tableName="bank" failed: table is being dropped
E191217 17:31:04.754402 172040 sql/lease.go:1194  background lease renewal for tableID=53 tableName="bank" failed: table is being dropped
E191217 17:31:04.862610 172057 sql/lease.go:1194  background lease renewal for tableID=53 tableName="bank" failed: table is being dropped
E191217 17:31:04.967536 172076 sql/lease.go:1194  background lease renewal for tableID=53 tableName="bank" failed: table is being dropped
E191217 17:31:05.069155 172060 sql/lease.go:1194  background lease renewal for tableID=53 tableName="bank" failed: table is being dropped
I191217 17:31:05.178939 117698 server/status/runtime.go:498  [n1] runtime stats: 2.2 GiB RSS, 761 goroutines, 111 MiB/62 MiB/204 MiB GO alloc/idle/total, 281 MiB/326 MiB CGO alloc/total, 787.5 CGO/sec, 173.4/13.6 %(u/s)time, 0.9 %gc (6x), 4.8 MiB/4.8 MiB (r/w)net
E191217 17:31:05.300711 172099 sql/lease.go:1194  background lease renewal for tableID=53 tableName="bank" failed: table is being dropped
E191217 17:31:05.573781 172151 sql/lease.go:1194  background lease renewal for tableID=53 tableName="bank" failed: table is being dropped
E191217 17:31:05.835404 172218 sql/lease.go:1194  background lease renewal for tableID=53 tableName="bank" failed: table is being dropped
E191217 17:31:05.961724 172147 sql/lease.go:1194  background lease renewal for tableID=53 tableName="bank" failed: table is being dropped
E191217 17:31:06.100651 172228 sql/lease.go:1194  background lease renewal for tableID=53 tableName="bank" failed: table is being dropped
E191217 17:31:06.212820 172172 sql/lease.go:1194  background lease renewal for tableID=53 tableName="bank" failed: table is being dropped
E191217 17:31:06.431140 172297 sql/lease.go:1194  background lease renewal for tableID=53 tableName="bank" failed: table is being dropped
E191217 17:31:06.502178 172347 sql/lease.go:1194  background lease renewal for tableID=53 tableName="bank" failed: table is being dropped
I191217 17:31:06.892594 118646 server/status/runtime.go:498  [n2] runtime stats: 2.2 GiB RSS, 765 goroutines, 112 MiB/61 MiB/204 MiB GO alloc/idle/total, 281 MiB/326 MiB CGO alloc/total, 787.1 CGO/sec, 173.1/12.5 %(u/s)time, 0.9 %gc (6x), 4.8 MiB/4.8 MiB (r/w)net
I191217 17:31:07.205626 122001 sql/sqlbase/structured.go:1425  [n1,client=127.0.0.1:42394,user=root,scExec] publish: descID=53 (bank) version=3 mtime=1970-01-01 00:00:00 +0000 UTC
W191217 17:31:07.519020 122001 sql/exec_util.go:1112  [n1,client=127.0.0.1:42394,user=root] error executing schema change: job with ID 512927886517698561 does not exist
I191217 17:31:07.544838 117442 util/stop/stopper.go:539  quiescing
I191217 17:31:07.699922 172566 util/stop/stopper.go:539  quiescing
I191217 17:31:07.700097 172568 util/stop/stopper.go:539  quiescing
I191217 17:31:07.702382 172567 util/stop/stopper.go:539  quiescing
W191217 17:31:07.717040 122270 storage/raft_transport.go:637  [n2] while processing outgoing Raft queue to node 3: EOF:
W191217 17:31:07.720062 119034 storage/raft_transport.go:637  [n3] while processing outgoing Raft queue to node 1: rpc error: code = Canceled desc = grpc: the client connection is closing:
W191217 17:31:07.720508 122076 storage/raft_transport.go:637  [n3] while processing outgoing Raft queue to node 2: rpc error: code = Canceled desc = grpc: the client connection is closing:
W191217 17:31:07.721179 122313 storage/raft_transport.go:637  [n3] while processing outgoing Raft queue to node 2: rpc error: code = Canceled desc = grpc: the client connection is closing:
W191217 17:31:07.722889 118671 gossip/gossip.go:1517  [n3] no incoming or outgoing connections
W191217 17:31:07.724432 119120 storage/raft_transport.go:637  [n2] while processing outgoing Raft queue to node 1: EOF:
W191217 17:31:07.734943 119071 storage/raft_transport.go:637  [n1] while processing outgoing Raft queue to node 2: rpc error: code = Canceled desc = grpc: the client connection is closing:
W191217 17:31:07.737886 119079 storage/raft_transport.go:637  [n3] while processing outgoing Raft queue to node 1: EOF:
W191217 17:31:07.740848 118953 storage/raft_transport.go:637  [n1] while processing outgoing Raft queue to node 3: rpc error: code = Canceled desc = grpc: the client connection is closing:
W191217 17:31:07.734478 119043 storage/raft_transport.go:637  [n1] while processing outgoing Raft queue to node 3: rpc error: code = Canceled desc = grpc: the client connection is closing:
W191217 17:31:07.745381 119186 storage/raft_transport.go:637  [n1] while processing outgoing Raft queue to node 2: EOF:
W191217 17:31:07.751420 119148 storage/raft_transport.go:637  [n2] while processing outgoing Raft queue to node 1: EOF:
I191217 17:31:07.757611 172556 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n1: failed to connect to n1 at 127.0.0.1:44127: context canceled
I191217 17:31:07.758938 172557 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n3: failed to connect to n3 at 127.0.0.1:40541: context canceled
W191217 17:31:07.766687 118497 gossip/gossip.go:1517  [n2] no incoming or outgoing connections
W191217 17:31:07.769363 172372 storage/intentresolver/intent_resolver.go:818  failed to gc transaction record: could not GC completed transaction anchored at /Table/SystemConfigSpan/Start: node unavailable; try another peer
W191217 17:31:07.774556 122347 storage/raft_transport.go:637  [n2] while processing outgoing Raft queue to node 3: rpc error: code = Canceled desc = grpc: the client connection is closing:
I191217 17:31:07.796195 172505 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n2: failed to connect to n2 at 127.0.0.1:35659: context canceled
I191217 17:31:07.805069 172506 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n3: failed to connect to n3 at 127.0.0.1:40541: context canceled
I191217 17:31:07.825011 118934 server/status/runtime.go:498  [n3] runtime stats: 2.1 GiB RSS, 408 goroutines, 85 MiB/84 MiB/204 MiB GO alloc/idle/total(stale), 259 MiB/311 MiB CGO alloc/total, 763.3 CGO/sec, 174.1/12.9 %(u/s)time, 0.7 %gc (6x), 4.8 MiB/4.8 MiB (r/w)net
W191217 17:31:07.829547 118647 ts/db.go:194  [n2,ts-poll] error writing time series data: node unavailable; try another peer
I191217 17:31:07.847632 172569 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n2: failed to connect to n2 at 127.0.0.1:35659: context canceled
I191217 17:31:07.850185 172570 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n1: failed to connect to n1 at 127.0.0.1:44127: context canceled
E191217 17:31:07.944140 117613 jobs/registry.go:358  error while adopting jobs: adopt-job: node unavailable; try another peer
I191217 17:31:07.957373 117709 kv/transport_race.go:108  transport race promotion: ran 293 iterations on up to 48119 requests
E191217 17:31:08.024460 172462 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
--- FAIL: TestBackupRestoreWithConcurrentWrites (333.34s)
    backup_test.go:2007: error executing 'DROP TABLE data.bank': pq: job with ID 512927886517698561 does not exist
```

<details><summary>Repro</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestBackupRestoreWithConcurrentWrites PKG=./pkg/ccl/backupccl TESTTIMEOUT=5m STRESSFLAGS=-timeout 5m' 2>&1
```

<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(ccl/backupccl).TestBackupRestoreWithConcurrentWrites failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1840228&tab=buildLog) on [master@1320e1360a2cbed8df8fbaa5e7100155999e534e](https://github.com/cockroachdb/cockroach/commits/1320e1360a2cbed8df8fbaa5e7100155999e534e):

Fatal error:
```
panic: event RetriableErr{CanAutoRetry:false, IsCommit:false} inappropriate in current state NoTxn{} [recovered]
	panic: panic while executing 1 statements: UPDATE _._ SET _ = $1 WHERE _ = $2; caused by event RetriableErr{CanAutoRetry:false, IsCommit:false} inappropriate in current state NoTxn{}

```

Stack:
```
goroutine 134277 [running]:
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).closeWrapper(0xc006d7e000, 0x464ba80, 0xc005852ac0, 0x3adca20, 0xc0003b4e80)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:810 +0x3f9
github.com/cockroachdb/cockroach/pkg/sql.(*Server).ServeConn.func1(0xc006d7e000, 0x464ba80, 0xc005852ac0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:474 +0x61
panic(0x3adca20, 0xc0003b4e80)
	/usr/local/go/src/runtime/panic.go:679 +0x1b2
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).txnStateTransitionsApplyWrapper(0xc006d7e000, 0x45d2880, 0xc0003b4e60, 0x3adb440, 0xc009d80ed0, 0x7f0066c21478, 0xc001002680, 0x357, 0x0, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:2085 +0xc04
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execCmd(0xc006d7e000, 0x464ba80, 0xc005852ac0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1497 +0x952
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).run(0xc006d7e000, 0x464ba80, 0xc005852ac0, 0xc005513540, 0x5400, 0x15000, 0xc0055135d8, 0xc001e83c10, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1297 +0x1f2
github.com/cockroachdb/cockroach/pkg/sql.(*Server).ServeConn(0xc000b57180, 0x464ba80, 0xc005852ac0, 0xc006d7e000, 0x5400, 0x15000, 0xc0055135d8, 0xc001e83c10, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:476 +0x104
github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*conn).processCommandsAsync.func1(0xc002d82442, 0xc005a86980, 0x464ba80, 0xc005852ac0, 0xc001e83c10, 0xc000b57180, 0xc001002000, 0x466e640, 0xc00491eba0, 0xc0056e4ae0, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn.go:594 +0x2e8
created by github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*conn).processCommandsAsync
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn.go:526 +0x17e

```

<details><summary>Log preceding fatal error</summary><p>

```
I200331 16:55:57.664457 133372 kv/kvserver/replica_command.go:397  [n1,split,s1,r32/1:/{Table/36-Max}] initiating a split of this range at key /Table/53 [r33] (zone config)
I200331 16:55:57.689034 129562 workload/workloadsql/dataload.go:140  imported bank (0s, 10 rows)
I200331 16:55:57.691400 129562 workload/workloadsql/workloadsql.go:113  starting 9 splits
I200331 16:55:57.692495 132970 kv/kvserver/replica_command.go:397  [n1,s1,r33/1:/{Table/53-Max}] initiating a split of this range at key /Table/53/1/5 [r34] (manual)
I200331 16:55:57.702677 133404 kv/kvserver/replica_command.go:397  [n3,s3,r34/3:/{Table/53/1/5-Max}] initiating a split of this range at key /Table/53/1/8 [r43] (manual)
I200331 16:55:57.721108 132970 kv/kvserver/replica_command.go:397  [n1,s1,r43/1:/{Table/53/1/8-Max}] initiating a split of this range at key /Table/53/1/9 [r35] (manual)
I200331 16:55:57.721899 133520 kv/kvserver/replica_command.go:397  [n1,s1,r33/1:/Table/53{-/1/5}] initiating a split of this range at key /Table/53/1/3 [r36] (manual)
I200331 16:55:57.736478 133520 kv/kvserver/replica_command.go:397  [n1,s1,r36/1:/Table/53/1/{3-5}] initiating a split of this range at key /Table/53/1/4 [r37] (manual)
I200331 16:55:57.741558 133409 kv/kvserver/replica_command.go:397  [n3,s3,r34/3:/Table/53/1/{5-8}] initiating a split of this range at key /Table/53/1/7 [r44] (manual)
I200331 16:55:57.744183 133520 kv/kvserver/replica_command.go:397  [n1,s1,r33/1:/Table/53{-/1/3}] initiating a split of this range at key /Table/53/1/2 [r38] (manual)
I200331 16:55:57.753150 133073 kv/kvserver/replica_command.go:397  [n1,s1,r33/1:/Table/53{-/1/2}] initiating a split of this range at key /Table/53/1/1 [r39] (manual)
I200331 16:55:57.761129 131174 gossip/gossip.go:1527  [n3] node has connected to cluster via gossip
I200331 16:55:57.761277 131174 kv/kvserver/stores.go:266  [n3] wrote 2 node addresses to persistent storage
I200331 16:55:57.761423 134149 kv/kvserver/replica_command.go:397  [n3,s3,r34/3:/Table/53/1/{5-7}] initiating a split of this range at key /Table/53/1/6 [r45] (manual)
I200331 16:55:57.785644 129562 testutils/testcluster/testcluster.go:760  WaitForFullReplication took: 13.594739ms
I200331 16:55:57.994716 130865 gossip/gossip.go:1527  [n2] node has connected to cluster via gossip
I200331 16:55:57.994859 130865 kv/kvserver/stores.go:266  [n2] wrote 2 node addresses to persistent storage
I200331 16:55:58.167786 135115 jobs/registry.go:797  [n1] job 542649083993423873: stepping through state running with error <nil>
I200331 16:55:58.191453 135304 ccl/storageccl/export.go:102  [n1,s1,r43/1:/Table/53/1/{8-9}] export [/Table/53/1/8,/Table/53/1/9)
I200331 16:55:58.191549 135301 ccl/storageccl/export.go:102  [n1,s1,r33/1:/Table/53{-/1/1}] export [/Table/53/1,/Table/53/1/1)
I200331 16:55:58.192039 135257 ccl/storageccl/export.go:102  [n3,s3,r37/3:/Table/53/1/{4-5}] export [/Table/53/1/4,/Table/53/1/5)
I200331 16:55:58.192518 135280 ccl/storageccl/export.go:102  [n3,s3,r45/3:/Table/53/1/{6-7}] export [/Table/53/1/6,/Table/53/1/7)
I200331 16:55:58.192998 135306 ccl/storageccl/export.go:102  [n1,s1,r36/1:/Table/53/1/{3-4}] export [/Table/53/1/3,/Table/53/1/4)
I200331 16:55:58.193216 135284 ccl/storageccl/export.go:102  [n3,s3,r35/3:/{Table/53/1/9-Max}] export [/Table/53/1/9,/Table/53/2)
I200331 16:55:58.193690 135240 ccl/storageccl/export.go:102  [n2,s2,r44/2:/Table/53/1/{7-8}] export [/Table/53/1/7,/Table/53/1/8)
I200331 16:55:58.196574 135256 ccl/storageccl/export.go:102  [n3,s3,r34/3:/Table/53/1/{5-6}] export [/Table/53/1/5,/Table/53/1/6)
I200331 16:55:58.197111 135305 ccl/storageccl/export.go:102  [n1,s1,r39/1:/Table/53/1/{1-2}] export [/Table/53/1/1,/Table/53/1/2)
I200331 16:55:58.199828 135151 ccl/storageccl/export.go:102  [n2,s2,r38/2:/Table/53/1/{2-3}] export [/Table/53/1/2,/Table/53/1/3)
I200331 16:55:58.250198 135115 jobs/registry.go:797  [n1] job 542649083993423873: stepping through state succeeded with error <nil>
I200331 16:55:58.272124 135115 jobs/registry.go:952  [n1] job 542649083993423873: status succeeded after adoption finished
I200331 16:55:58.340210 133520 sql/table.go:942  [n1,client=127.0.0.1:44800,hostssl,user=root] queued new schema change job for table 53, mutation 0
I200331 16:55:58.340261 133520 sql/sqlbase/structured.go:1554  [n1,client=127.0.0.1:44800,hostssl,user=root] publish: descID=53 (bank) version=2 mtime=1970-01-01 00:00:00 +0000 UTC
I200331 16:55:58.349834 133520 sql/event_log.go:132  [n1,client=127.0.0.1:44800,hostssl,user=root] Event: "drop_table", target: 53, info: {TableName:data.public.bank Statement:DROP TABLE data.public.bank User:root CascadeDroppedViews:[]}
I200331 16:55:58.350017 133520 jobs/registry.go:285  [n1,client=127.0.0.1:44800,hostssl,user=root] scheduled jobs [542649084604612609]
I200331 16:55:58.352964 135681 kv/kvserver/replica_command.go:663  [n1,merge,s1,r33/1:/Table/53{-/1/1}] initiating a merge of r39:/Table/53/1/{1-2} [(n1,s1):1, (n2,s2):2, (n3,s3):3, next=4, gen=9] into this range (lhs+rhs has (size=4.0 KiB+4.0 KiB=8.0 KiB qps=0.00+0.00=0.00qps) below threshold (size=128 MiB, qps=1250.00))
I200331 16:55:58.362391 129729 kv/kvserver/store_remove_replica.go:122  [n1,s1,r33/1:/Table/53{-/1/1}] removing replica r39/1
I200331 16:55:58.363563 131287 kv/kvserver/store_remove_replica.go:122  [n3,s3,r33/3:/Table/53{-/1/1}] removing replica r39/3
I200331 16:55:58.363616 131001 kv/kvserver/store_remove_replica.go:122  [n2,s2,r33/2:/Table/53{-/1/1}] removing replica r39/2
I200331 16:55:58.366995 135957 kv/kvserver/replica_command.go:663  [n1,merge,s1,r43/1:/Table/53/1/{8-9}] initiating a merge of r35:/{Table/53/1/9-Max} [(n1,s1):1, (n2,s2):2, (n3,s3):3, next=4, gen=8] into this range (lhs+rhs has (size=5.3 KiB+6.5 KiB=12 KiB qps=0.00+0.00=0.00qps) below threshold (size=128 MiB, qps=1250.00))
I200331 16:55:58.368163 135914 kv/kvserver/replica_command.go:663  [n3,merge,s3,r37/3:/Table/53/1/{4-5}] initiating a merge of r34:/Table/53/1/{5-6} [(n1,s1):1, (n2,s2):2, (n3,s3):3, next=4, gen=9] into this range (lhs+rhs has (size=6.7 KiB+7.2 KiB=14 KiB qps=0.00+0.00=0.00qps) below threshold (size=128 MiB, qps=1250.00))
I200331 16:55:58.370750 129601 jobs/jobs.go:743  job 542649084604612609: adopting
I200331 16:55:58.376580 129601 jobs/registry.go:1178  job 542649084604612609: resuming execution
I200331 16:55:58.378738 136014 jobs/registry.go:797  [n1] job 542649084604612609: stepping through state running with error <nil>
I200331 16:55:58.379664 136014 sql/schema_changer.go:445  [n1,job=542649084604612609,scExec] schema change on bank (53 v2) mutation 0 starting execution...
I200331 16:55:58.381889 136014 sql/sqlbase/structured.go:1554  [n1,job=542649084604612609,scExec] publish: descID=53 (bank) version=3 mtime=1970-01-01 00:00:00 +0000 UTC
I200331 16:55:58.383949 129693 kv/kvserver/store_remove_replica.go:122  [n1,s1,r43/1:/Table/53/1/{8-9}] removing replica r35/1
I200331 16:55:58.384119 131299 kv/kvserver/store_remove_replica.go:122  [n3,s3,r43/3:/Table/53/1/{8-9}] removing replica r35/3
I200331 16:55:58.385271 130989 kv/kvserver/store_remove_replica.go:122  [n2,s2,r43/2:/Table/53/1/{8-9}] removing replica r35/2
E200331 16:55:58.385910 134277 sql/conn_executor.go:796  [n2,client=127.0.0.1:47084,hostssl,user=root] a SQL panic has occurred while executing "UPDATE data.bank SET payload = $1 WHERE id = $2": event RetriableErr{CanAutoRetry:false, IsCommit:false} inappropriate in current state NoTxn{}
E200331 16:55:58.386008 134277 util/log/crash_reporting.go:208  [n2,client=127.0.0.1:47084,hostssl,user=root] a panic has occurred!

```

</p></details>

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestBackupRestoreWithConcurrentWrites PKG=./pkg/ccl/backupccl TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestBackupRestoreWithConcurrentWrites.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(ccl/backupccl).TestBackupRestoreWithConcurrentWrites failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1840480&tab=buildLog) on [master@39754d8e355603f17c32d15314f3753fea022a4c](https://github.com/cockroachdb/cockroach/commits/39754d8e355603f17c32d15314f3753fea022a4c):

Fatal error:
```
panic: event RetriableErr{CanAutoRetry:false, IsCommit:false} inappropriate in current state NoTxn{} [recovered]
	panic: panic while executing 1 statements: UPDATE _._ SET _ = $1 WHERE _ = $2; caused by event RetriableErr{CanAutoRetry:false, IsCommit:false} inappropriate in current state NoTxn{}

```

Stack:
```
goroutine 170794 [running]:
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).closeWrapper(0xc000a24000, 0x6ea5f00, 0xc0041c0040, 0x5c4f900, 0xc0054a39a0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:810 +0x4e1
github.com/cockroachdb/cockroach/pkg/sql.(*Server).ServeConn.func1(0xc000a24000, 0x6ea5f00, 0xc0041c0040)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:474 +0x6f
panic(0x5c4f900, 0xc0054a39a0)
	/usr/local/go/src/runtime/panic.go:679 +0x1b2
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).txnStateTransitionsApplyWrapper(0xc000a24000, 0x6e28840, 0xc0054a3980, 0x5c4e320, 0xc0039fa780, 0x7fa8fdea33d8, 0xc002764680, 0xf3, 0x0, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:2085 +0x16b5
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execCmd(0xc000a24000, 0x6ea5f00, 0xc0041c0040, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1497 +0xa77
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).run(0xc000a24000, 0x6ea5f00, 0xc0041c0040, 0xc0029f9790, 0x5400, 0x15000, 0xc0029f9830, 0xc005086a30, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1297 +0x484
github.com/cockroachdb/cockroach/pkg/sql.(*Server).ServeConn(0xc00089d700, 0x6ea5f00, 0xc0041c0040, 0xc000a24000, 0x5400, 0x15000, 0xc0029f9830, 0xc005086a30, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:476 +0x132
github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*conn).processCommandsAsync.func1(0xc0033abd62, 0xc002290120, 0x6ea5f00, 0xc0041c0040, 0xc005086a30, 0xc00089d700, 0xc002764000, 0x6eca240, 0xc000b53f80, 0xc001faede0, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn.go:594 +0x36c
created by github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*conn).processCommandsAsync
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn.go:526 +0x1ea

```

<details><summary>Log preceding fatal error</summary><p>

```
I200331 18:38:19.017577 170257 kv/kvserver/replica_command.go:397  [n1,s1,r33/1:/Table/53{-/1/2}] initiating a split of this range at key /Table/53/1/1 [r37] (manual)
I200331 18:38:19.498123 169691 kv/kvserver/replica_command.go:397  [n1,s1,r43/1:/{Table/53/1/8-Max}] initiating a split of this range at key /Table/53/1/9 [r38] (manual)
I200331 18:38:19.613491 170552 kv/kvserver/replica_command.go:397  [n2,s2,r34/2:/Table/53/1/{5-8}] initiating a split of this range at key /Table/53/1/7 [r44] (manual)
I200331 18:38:20.196420 164844 server/status/runtime.go:498  [n1] runtime stats: 1.3 GiB RSS, 1149 goroutines, 79 MiB/91 MiB/134 MiB GO alloc/idle/total, 116 MiB/159 MiB CGO alloc/total, 2363.7 CGO/sec, 201.4/15.6 %(u/s)time, 0.5 %gc (10x), 3.7 MiB/3.7 MiB (r/w)net
I200331 18:38:20.251838 170656 kv/kvserver/replica_command.go:397  [n2,s2,r34/2:/Table/53/1/{5-7}] initiating a split of this range at key /Table/53/1/6 [r45] (manual)
I200331 18:38:20.852936 164699 testutils/testcluster/testcluster.go:760  WaitForFullReplication took: 178.744516ms
I200331 18:38:24.541627 171258 jobs/registry.go:797  [n1] job 542669221923061761: stepping through state running with error <nil>
I200331 18:38:24.614609 166229 server/status/runtime.go:498  [n2] runtime stats: 1.3 GiB RSS, 773 goroutines, 65 MiB/103 MiB/128 MiB GO alloc/idle/total, 95 MiB/142 MiB CGO alloc/total, 1817.1 CGO/sec, 202.6/16.4 %(u/s)time, 0.4 %gc (10x), 3.9 MiB/3.9 MiB (r/w)net
I200331 18:38:25.578812 166662 server/status/runtime.go:498  [n3] runtime stats: 1.3 GiB RSS, 774 goroutines, 58 MiB/109 MiB/130 MiB GO alloc/idle/total, 95 MiB/140 MiB CGO alloc/total, 1700.4 CGO/sec, 201.8/16.1 %(u/s)time, 0.3 %gc (10x), 3.9 MiB/3.9 MiB (r/w)net
I200331 18:38:25.769824 171471 ccl/storageccl/export.go:102  [n1,s1,r33/1:/Table/53{-/1/1}] export [/Table/53/1,/Table/53/1/1)
I200331 18:38:25.818970 171466 ccl/storageccl/export.go:102  [n1,s1,r43/1:/Table/53/1/{8-9}] export [/Table/53/1/8,/Table/53/1/9)
I200331 18:38:25.825177 171468 ccl/storageccl/export.go:102  [n1,s1,r38/1:/{Table/53/1/9-Max}] export [/Table/53/1/9,/Table/53/2)
I200331 18:38:25.840705 171477 ccl/storageccl/export.go:102  [n2,s2,r44/2:/Table/53/1/{7-8}] export [/Table/53/1/7,/Table/53/1/8)
I200331 18:38:25.843772 171476 ccl/storageccl/export.go:102  [n2,s2,r37/2:/Table/53/1/{1-2}] export [/Table/53/1/1,/Table/53/1/2)
I200331 18:38:25.846111 171391 ccl/storageccl/export.go:102  [n3,s3,r45/3:/Table/53/1/{6-7}] export [/Table/53/1/6,/Table/53/1/7)
I200331 18:38:25.848757 171392 ccl/storageccl/export.go:102  [n3,s3,r36/3:/Table/53/1/{2-3}] export [/Table/53/1/2,/Table/53/1/3)
I200331 18:38:25.850882 171390 ccl/storageccl/export.go:102  [n3,s3,r35/3:/Table/53/1/{3-4}] export [/Table/53/1/3,/Table/53/1/4)
I200331 18:38:25.881894 171478 ccl/storageccl/export.go:102  [n2,s2,r34/2:/Table/53/1/{5-6}] export [/Table/53/1/5,/Table/53/1/6)
I200331 18:38:25.910188 171411 ccl/storageccl/export.go:102  [n2,s2,r53/2:/Table/53/1/{4-5}] export [/Table/53/1/4,/Table/53/1/5)
I200331 18:38:27.618887 171258 jobs/registry.go:797  [n1] job 542669221923061761: stepping through state succeeded with error <nil>
I200331 18:38:28.080335 171258 jobs/registry.go:952  [n1] job 542669221923061761: status succeeded after adoption finished
I200331 18:38:30.192907 164844 server/status/runtime.go:498  [n1] runtime stats: 1.3 GiB RSS, 778 goroutines, 61 MiB/107 MiB/133 MiB GO alloc/idle/total, 98 MiB/145 MiB CGO alloc/total, 987.6 CGO/sec, 200.5/17.2 %(u/s)time, 0.5 %gc (9x), 4.0 MiB/4.0 MiB (r/w)net
I200331 18:38:30.561930 170257 sql/table.go:942  [n1,client=127.0.0.1:55114,hostssl,user=root] queued new schema change job for table 53, mutation 0
I200331 18:38:30.562245 170257 sql/sqlbase/structured.go:1554  [n1,client=127.0.0.1:55114,hostssl,user=root] publish: descID=53 (bank) version=2 mtime=1970-01-01 00:00:00 +0000 UTC
I200331 18:38:31.013736 170257 sql/event_log.go:132  [n1,client=127.0.0.1:55114,hostssl,user=root] Event: "drop_table", target: 53, info: {TableName:data.public.bank Statement:DROP TABLE data.public.bank User:root CascadeDroppedViews:[]}
I200331 18:38:31.014281 170257 jobs/registry.go:285  [n1,client=127.0.0.1:55114,hostssl,user=root] scheduled jobs [542669243577630721]
I200331 18:38:31.395160 172056 kv/kvserver/replica_command.go:663  [n3,merge,s3,r36/3:/Table/53/1/{2-3}] initiating a merge of r35:/Table/53/1/{3-4} [(n1,s1):1, (n2,s2):2, (n3,s3):3, next=4, gen=8] into this range (lhs+rhs has (size=1.9 KiB+2.0 KiB=3.9 KiB qps=15.65+10.54=26.18qps) below threshold (size=128 MiB, qps=1250.00))
I200331 18:38:31.421118 171907 kv/kvserver/replica_command.go:663  [n2,merge,s2,r53/2:/Table/53/1/{4-5}] initiating a merge of r34:/Table/53/1/{5-6} [(n1,s1):1, (n2,s2):2, (n3,s3):3, next=4, gen=9] into this range (lhs+rhs has (size=1.3 KiB+2.0 KiB=3.3 KiB qps=6.22+6.96=13.18qps) below threshold (size=128 MiB, qps=1250.00))
I200331 18:38:31.568148 172156 kv/kvserver/replica_command.go:663  [n1,merge,s1,r33/1:/Table/53{-/1/1}] initiating a merge of r37:/Table/53/1/{1-2} [(n1,s1):1, (n2,s2):2, (n3,s3):3, next=4, gen=9] into this range (lhs+rhs has (size=1.9 KiB+2.7 KiB=4.6 KiB qps=5.83+8.35=14.18qps) below threshold (size=128 MiB, qps=1250.00))
I200331 18:38:32.018723 164926 jobs/jobs.go:743  job 542669243577630721: adopting
I200331 18:38:32.450870 164926 jobs/registry.go:1178  job 542669243577630721: resuming execution
I200331 18:38:32.723223 166301 kv/kvserver/store_remove_replica.go:122  [n2,s2,r53/2:/Table/53/1/{4-5}] removing replica r34/2
I200331 18:38:32.733105 164819 kv/kvserver/store_remove_replica.go:122  [n1,s1,r53/1:/Table/53/1/{4-5}] removing replica r34/1
I200331 18:38:32.740002 166573 kv/kvserver/store_remove_replica.go:122  [n3,s3,r53/3:/Table/53/1/{4-5}] removing replica r34/3
I200331 18:38:32.724203 172462 jobs/registry.go:797  [n1] job 542669243577630721: stepping through state running with error <nil>
I200331 18:38:32.785689 172462 sql/schema_changer.go:445  [n1,job=542669243577630721,scExec] schema change on bank (53 v2) mutation 0 starting execution...
I200331 18:38:32.813494 164763 kv/kvserver/store_remove_replica.go:122  [n1,s1,r33/1:/Table/53{-/1/1}] removing replica r37/1
I200331 18:38:32.820806 166282 kv/kvserver/store_remove_replica.go:122  [n2,s2,r33/2:/Table/53{-/1/1}] removing replica r37/2
I200331 18:38:32.835779 166594 kv/kvserver/store_remove_replica.go:122  [n3,s3,r36/3:/Table/53/1/{2-3}] removing replica r35/3
I200331 18:38:32.839016 166582 kv/kvserver/store_remove_replica.go:122  [n3,s3,r33/3:/Table/53{-/1/1}] removing replica r37/3
I200331 18:38:32.883856 164799 kv/kvserver/store_remove_replica.go:122  [n1,s1,r36/1:/Table/53/1/{2-3}] removing replica r35/1
I200331 18:38:32.903463 166272 kv/kvserver/store_remove_replica.go:122  [n2,s2,r36/2:/Table/53/1/{2-3}] removing replica r35/2
I200331 18:38:33.135322 172561 kv/kvserver/replica_command.go:663  [n1,merge,s1,r43/1:/Table/53/1/{8-9}] initiating a merge of r38:/{Table/53/1/9-Max} [(n1,s1):1, (n2,s2):2, (n3,s3):3, next=4, gen=8] into this range (lhs+rhs has (size=2.6 KiB+2.0 KiB=4.6 KiB qps=0.63+0.79=1.42qps) below threshold (size=128 MiB, qps=1250.00))
I200331 18:38:33.151183 172462 sql/sqlbase/structured.go:1554  [n1,job=542669243577630721,scExec] publish: descID=53 (bank) version=3 mtime=1970-01-01 00:00:00 +0000 UTC
I200331 18:38:33.357967 172543 kv/kvserver/replica_command.go:663  [n3,merge,s3,r45/3:/Table/53/1/{6-7}] initiating a merge of r44:/Table/53/1/{7-8} [(n1,s1):1, (n2,s2):2, (n3,s3):3, next=4, gen=8] into this range (lhs+rhs has (size=1.4 KiB+3.0 KiB=4.4 KiB qps=0.00+0.00=0.00qps) below threshold (size=128 MiB, qps=1250.00))
I200331 18:38:33.650105 164806 kv/kvserver/store_remove_replica.go:122  [n1,s1,r43/1:/Table/53/1/{8-9}] removing replica r38/1
I200331 18:38:33.661943 166288 kv/kvserver/store_remove_replica.go:122  [n2,s2,r43/2:/Table/53/1/{8-9}] removing replica r38/2
I200331 18:38:33.665265 166579 kv/kvserver/store_remove_replica.go:122  [n3,s3,r43/3:/Table/53/1/{8-9}] removing replica r38/3
E200331 18:38:33.871983 170794 sql/conn_executor.go:796  [n3,client=127.0.0.1:51590,hostssl,user=root] a SQL panic has occurred while executing "UPDATE data.bank SET payload = $1 WHERE id = $2": event RetriableErr{CanAutoRetry:false, IsCommit:false} inappropriate in current state NoTxn{}
E200331 18:38:33.872900 170794 util/log/crash_reporting.go:208  [n3,client=127.0.0.1:51590,hostssl,user=root] a panic has occurred!

```

</p></details>

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestBackupRestoreWithConcurrentWrites PKG=./pkg/ccl/backupccl TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestBackupRestoreWithConcurrentWrites.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(ccl/backupccl).TestBackupRestoreWithConcurrentWrites failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1840639&tab=buildLog) on [master@98a351e518bb1f2834db556ee9d56f4158364fc2](https://github.com/cockroachdb/cockroach/commits/98a351e518bb1f2834db556ee9d56f4158364fc2):

Fatal error:
```
panic: event RetriableErr{CanAutoRetry:false, IsCommit:false} inappropriate in current state NoTxn{} [recovered]
	panic: panic while executing 1 statements: UPDATE _._ SET _ = $1 WHERE _ = $2; caused by event RetriableErr{CanAutoRetry:false, IsCommit:false} inappropriate in current state NoTxn{}

```

Stack:
```
goroutine 134093 [running]:
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).closeWrapper(0xc006a4e000, 0x464c9a0, 0xc0045ddf80, 0x3add6a0, 0xc009d5b680)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:810 +0x3f9
github.com/cockroachdb/cockroach/pkg/sql.(*Server).ServeConn.func1(0xc006a4e000, 0x464c9a0, 0xc0045ddf80)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:474 +0x61
panic(0x3add6a0, 0xc009d5b680)
	/usr/local/go/src/runtime/panic.go:679 +0x1b2
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).txnStateTransitionsApplyWrapper(0xc006a4e000, 0x45d3760, 0xc009d5b660, 0x3adc0c0, 0xc0009305a0, 0x7f29ff423450, 0xc004621880, 0x147, 0x0, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:2085 +0xc04
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execCmd(0xc006a4e000, 0x464c9a0, 0xc0045ddf80, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1497 +0x952
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).run(0xc006a4e000, 0x464c9a0, 0xc0045ddf80, 0xc008c32100, 0x5400, 0x15000, 0xc008c32198, 0xc0073e7020, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1297 +0x1f2
github.com/cockroachdb/cockroach/pkg/sql.(*Server).ServeConn(0xc008b16000, 0x464c9a0, 0xc0045ddf80, 0xc006a4e000, 0x5400, 0x15000, 0xc008c32198, 0xc0073e7020, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:476 +0x104
github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*conn).processCommandsAsync.func1(0xc003d9e5a2, 0xc006b61320, 0x464c9a0, 0xc0045ddf80, 0xc0073e7020, 0xc008b16000, 0xc004621200, 0x466f560, 0xc00758e0f0, 0xc0077b7440, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn.go:594 +0x2e8
created by github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*conn).processCommandsAsync
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn.go:526 +0x17e

```

<details><summary>Log preceding fatal error</summary><p>

```
I200331 19:50:11.959777 132841 sql/event_log.go:132  [n1,client=127.0.0.1:54906,hostssl,user=root] Event: "create_database", target: 52, info: {DatabaseName:data Statement:CREATE DATABASE data User:root}
I200331 19:50:11.986171 132841 sql/event_log.go:132  [n1,client=127.0.0.1:54906,hostssl,user=root] Event: "create_table", target: 53, info: {TableName:data.public.bank Statement:CREATE TABLE IF NOT EXISTS bank (id INT8 PRIMARY KEY, balance INT8, payload STRING, FAMILY (id, balance, payload)) User:root}
I200331 19:50:11.992353 133206 kv/kvserver/replica_command.go:397  [n1,split,s1,r32/1:/{Table/36-Max}] initiating a split of this range at key /Table/53 [r33] (zone config)
I200331 19:50:11.997814 129414 gossip/gossip.go:1527  [n1] node has connected to cluster via gossip
I200331 19:50:11.997987 129414 kv/kvserver/stores.go:266  [n1] wrote 2 node addresses to persistent storage
I200331 19:50:12.005222 129040 workload/workloadsql/dataload.go:140  imported bank (0s, 10 rows)
I200331 19:50:12.008231 129040 workload/workloadsql/workloadsql.go:113  starting 9 splits
I200331 19:50:12.016970 132841 kv/kvserver/replica_command.go:397  [n1,s1,r33/1:/{Table/53-Max}] initiating a split of this range at key /Table/53/1/5 [r34] (manual)
I200331 19:50:12.040631 133711 kv/kvserver/replica_command.go:397  [n3,s3,r34/3:/{Table/53/1/5-Max}] initiating a split of this range at key /Table/53/1/8 [r43] (manual)
I200331 19:50:12.058168 133750 kv/kvserver/replica_command.go:397  [n1,s1,r33/1:/Table/53{-/1/5}] initiating a split of this range at key /Table/53/1/3 [r35] (manual)
I200331 19:50:12.084014 133752 kv/kvserver/replica_command.go:397  [n3,s3,r35/3:/Table/53/1/{3-5}] initiating a split of this range at key /Table/53/1/4 [r44] (manual)
I200331 19:50:12.086723 133802 kv/kvserver/replica_command.go:397  [n3,s3,r34/3:/Table/53/1/{5-8}] initiating a split of this range at key /Table/53/1/7 [r45] (manual)
I200331 19:50:12.116561 133863 kv/kvserver/replica_command.go:397  [n1,s1,r33/1:/Table/53{-/1/3}] initiating a split of this range at key /Table/53/1/2 [r36] (manual)
I200331 19:50:12.134358 133897 kv/kvserver/replica_command.go:397  [n3,s3,r34/3:/Table/53/1/{5-7}] initiating a split of this range at key /Table/53/1/6 [r46] (manual)
I200331 19:50:12.153692 133863 kv/kvserver/replica_command.go:397  [n1,s1,r33/1:/Table/53{-/1/2}] initiating a split of this range at key /Table/53/1/1 [r37] (manual)
I200331 19:50:12.276440 133794 kv/kvserver/replica_command.go:397  [n2,s2,r43/2:/{Table/53/1/8-Max}] initiating a split of this range at key /Table/53/1/9 [r53] (manual); delayed split for 0.2s to avoid Raft snapshot
I200331 19:50:12.322993 129040 testutils/testcluster/testcluster.go:760  WaitForFullReplication took: 18.510475ms
I200331 19:50:12.454433 130699 gossip/gossip.go:1527  [n2] node has connected to cluster via gossip
I200331 19:50:12.454688 130699 kv/kvserver/stores.go:266  [n2] wrote 2 node addresses to persistent storage
I200331 19:50:12.461776 134147 jobs/registry.go:797  [n1] job 542683340580519937: stepping through state running with error <nil>
I200331 19:50:12.516085 134508 ccl/storageccl/export.go:102  [n1,s1,r33/1:/Table/53{-/1/1}] export [/Table/53/1,/Table/53/1/1)
I200331 19:50:12.516222 134502 ccl/storageccl/export.go:102  [n1,s1,r45/1:/Table/53/1/{7-8}] export [/Table/53/1/7,/Table/53/1/8)
I200331 19:50:12.517614 134419 ccl/storageccl/export.go:102  [n1,s1,r53/1:/{Table/53/1/9-Max}] export [/Table/53/1/9,/Table/53/2)
I200331 19:50:12.519753 134519 ccl/storageccl/export.go:102  [n2,s2,r44/2:/Table/53/1/{4-5}] export [/Table/53/1/4,/Table/53/1/5)
I200331 19:50:12.521681 134517 ccl/storageccl/export.go:102  [n2,s2,r46/2:/Table/53/1/{6-7}] export [/Table/53/1/6,/Table/53/1/7)
I200331 19:50:12.522530 134518 ccl/storageccl/export.go:102  [n2,s2,r43/2:/Table/53/1/{8-9}] export [/Table/53/1/8,/Table/53/1/9)
I200331 19:50:12.523959 134510 ccl/storageccl/export.go:102  [n3,s3,r36/3:/Table/53/1/{2-3}] export [/Table/53/1/2,/Table/53/1/3)
I200331 19:50:12.525318 134501 ccl/storageccl/export.go:102  [n1,s1,r37/1:/Table/53/1/{1-2}] export [/Table/53/1/1,/Table/53/1/2)
I200331 19:50:12.526092 134520 ccl/storageccl/export.go:102  [n3,s3,r34/3:/Table/53/1/{5-6}] export [/Table/53/1/5,/Table/53/1/6)
I200331 19:50:12.539706 134509 ccl/storageccl/export.go:102  [n3,s3,r35/3:/Table/53/1/{3-4}] export [/Table/53/1/3,/Table/53/1/4)
I200331 19:50:12.579827 134147 jobs/registry.go:797  [n1] job 542683340580519937: stepping through state succeeded with error <nil>
I200331 19:50:12.599915 134147 jobs/registry.go:952  [n1] job 542683340580519937: status succeeded after adoption finished
I200331 19:50:12.634105 130991 gossip/gossip.go:1527  [n3] node has connected to cluster via gossip
I200331 19:50:12.634268 130991 kv/kvserver/stores.go:266  [n3] wrote 2 node addresses to persistent storage
I200331 19:50:12.684715 133821 sql/table.go:942  [n1,client=127.0.0.1:54912,hostssl,user=root] queued new schema change job for table 53, mutation 0
I200331 19:50:12.684815 133821 sql/sqlbase/structured.go:1554  [n1,client=127.0.0.1:54912,hostssl,user=root] publish: descID=53 (bank) version=2 mtime=1970-01-01 00:00:00 +0000 UTC
I200331 19:50:12.707923 133821 sql/event_log.go:132  [n1,client=127.0.0.1:54912,hostssl,user=root] Event: "drop_table", target: 53, info: {TableName:data.public.bank Statement:DROP TABLE data.public.bank User:root CascadeDroppedViews:[]}
I200331 19:50:12.708008 133821 jobs/registry.go:285  [n1,client=127.0.0.1:54912,hostssl,user=root] scheduled jobs [542683341396049921]
I200331 19:50:12.711383 134913 kv/kvserver/replica_command.go:663  [n1,merge,s1,r37/1:/Table/53/1/{1-2}] initiating a merge of r36:/Table/53/1/{2-3} [(n1,s1):1, (n2,s2):2, (n3,s3):3, next=4, gen=8] into this range (lhs+rhs has (size=3.1 KiB+2.3 KiB=5.3 KiB qps=0.00+0.00=0.00qps) below threshold (size=128 MiB, qps=1250.00))
I200331 19:50:12.723507 135019 kv/kvserver/replica_command.go:663  [n2,merge,s2,r44/2:/Table/53/1/{4-5}] initiating a merge of r34:/Table/53/1/{5-6} [(n1,s1):1, (n2,s2):2, (n3,s3):3, next=4, gen=9] into this range (lhs+rhs has (size=1.7 KiB+2.8 KiB=4.5 KiB qps=0.00+0.00=0.00qps) below threshold (size=128 MiB, qps=1250.00))
I200331 19:50:12.726012 129524 jobs/jobs.go:743  job 542683341396049921: adopting
I200331 19:50:12.731050 129524 jobs/registry.go:1178  job 542683341396049921: resuming execution
I200331 19:50:12.737862 135167 jobs/registry.go:797  [n1] job 542683341396049921: stepping through state running with error <nil>
I200331 19:50:12.738277 135167 sql/schema_changer.go:445  [n1,job=542683341396049921,scExec] schema change on bank (53 v2) mutation 0 starting execution...
I200331 19:50:12.741139 135167 sql/sqlbase/structured.go:1554  [n1,job=542683341396049921,scExec] publish: descID=53 (bank) version=3 mtime=1970-01-01 00:00:00 +0000 UTC
I200331 19:50:12.743709 129470 kv/kvserver/store_remove_replica.go:122  [n1,s1,r37/1:/Table/53/1/{1-2}] removing replica r36/1
I200331 19:50:12.744333 130896 kv/kvserver/store_remove_replica.go:122  [n3,s3,r37/3:/Table/53/1/{1-2}] removing replica r36/3
I200331 19:50:12.746091 130823 kv/kvserver/store_remove_replica.go:122  [n2,s2,r37/2:/Table/53/1/{1-2}] removing replica r36/2
E200331 19:50:12.750376 134093 sql/conn_executor.go:796  [n2,client=127.0.0.1:59864,hostssl,user=root] a SQL panic has occurred while executing "UPDATE data.bank SET payload = $1 WHERE id = $2": event RetriableErr{CanAutoRetry:false, IsCommit:false} inappropriate in current state NoTxn{}
E200331 19:50:12.750938 134093 util/log/crash_reporting.go:208  [n2,client=127.0.0.1:59864,hostssl,user=root] a panic has occurred!

```

</p></details>

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestBackupRestoreWithConcurrentWrites PKG=./pkg/ccl/backupccl TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestBackupRestoreWithConcurrentWrites.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(ccl/backupccl).TestBackupRestoreWithConcurrentWrites failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1840641&tab=buildLog) on [master@98a351e518bb1f2834db556ee9d56f4158364fc2](https://github.com/cockroachdb/cockroach/commits/98a351e518bb1f2834db556ee9d56f4158364fc2):

Fatal error:
```
panic: event RetriableErr{CanAutoRetry:false, IsCommit:false} inappropriate in current state NoTxn{} [recovered]
	panic: panic while executing 1 statements: UPDATE _._ SET _ = $1 WHERE _ = $2; caused by event RetriableErr{CanAutoRetry:false, IsCommit:false} inappropriate in current state NoTxn{}

```

Stack:
```
goroutine 163491 [running]:
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).closeWrapper(0xc000e64000, 0x6ea6b00, 0xc001c4e180, 0x5c501c0, 0xc006cde1a0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:810 +0x4e1
github.com/cockroachdb/cockroach/pkg/sql.(*Server).ServeConn.func1(0xc000e64000, 0x6ea6b00, 0xc001c4e180)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:474 +0x6f
panic(0x5c501c0, 0xc006cde1a0)
	/usr/local/go/src/runtime/panic.go:679 +0x1b2
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).txnStateTransitionsApplyWrapper(0xc000e64000, 0x6e29400, 0xc006cde180, 0x5c4ebe0, 0xc003402990, 0x7fe546500780, 0xc001e28180, 0x42, 0x0, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:2085 +0x16b5
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execCmd(0xc000e64000, 0x6ea6b00, 0xc001c4e180, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1497 +0xa77
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).run(0xc000e64000, 0x6ea6b00, 0xc001c4e180, 0xc0048c3a10, 0x5400, 0x15000, 0xc0048c3ab0, 0xc0025a9f60, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1297 +0x484
github.com/cockroachdb/cockroach/pkg/sql.(*Server).ServeConn(0xc00163b700, 0x6ea6b00, 0xc001c4e180, 0xc000e64000, 0x5400, 0x15000, 0xc0048c3ab0, 0xc0025a9f60, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:476 +0x132
github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*conn).processCommandsAsync.func1(0xc003a0f90d, 0xc006d53700, 0x6ea6b00, 0xc001c4e180, 0xc0025a9f60, 0xc00163b700, 0xc001e27b00, 0x6ecae40, 0xc0040c6690, 0xc002516300, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn.go:594 +0x36c
created by github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*conn).processCommandsAsync
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn.go:526 +0x1ea

```

<details><summary>Log preceding fatal error</summary><p>

```
I200331 20:02:59.982887 165236 ccl/storageccl/export.go:102  [n1,s1,r45/1:/Table/53/1/{7-8}] export [/Table/53/1/7,/Table/53/1/8)
I200331 20:02:59.993424 165238 ccl/storageccl/export.go:102  [n1,s1,r46/1:/Table/53/1/{6-7}] export [/Table/53/1/6,/Table/53/1/7)
I200331 20:02:59.993993 165244 ccl/storageccl/export.go:102  [n3,s3,r44/2:/{Table/53/1/9-Max}] export [/Table/53/1/9,/Table/53/2)
I200331 20:03:00.009891 165086 ccl/storageccl/export.go:102  [n3,s3,r35/2:/Table/53/1/{3-4}] export [/Table/53/1/3,/Table/53/1/4)
I200331 20:03:00.027369 165145 ccl/storageccl/export.go:102  [n2,s2,r36/3:/Table/53/1/{2-3}] export [/Table/53/1/2,/Table/53/1/3)
I200331 20:03:00.028040 165146 ccl/storageccl/export.go:102  [n2,s2,r43/3:/Table/53/1/{8-9}] export [/Table/53/1/8,/Table/53/1/9)
I200331 20:03:00.028424 165144 ccl/storageccl/export.go:102  [n2,s2,r63/3:/Table/53/1/{4-5}] export [/Table/53/1/4,/Table/53/1/5)
I200331 20:03:00.029735 165239 ccl/storageccl/export.go:102  [n1,s1,r33/1:/Table/53{-/1/1}] export [/Table/53/1,/Table/53/1/1)
I200331 20:03:00.033932 165245 ccl/storageccl/export.go:102  [n2,s2,r37/3:/Table/53/1/{1-2}] export [/Table/53/1/1,/Table/53/1/2)
I200331 20:03:00.034372 165147 ccl/storageccl/export.go:102  [n2,s2,r34/3:/Table/53/1/{5-6}] export [/Table/53/1/5,/Table/53/1/6)
I200331 20:03:00.506376 165114 jobs/registry.go:797  [n1] job 542685849444319233: stepping through state succeeded with error <nil>
I200331 20:03:00.644892 165114 jobs/registry.go:952  [n1] job 542685849444319233: status succeeded after adoption finished
I200331 20:03:01.731607 163089 sql/table.go:942  [n1,client=127.0.0.1:60110,hostssl,user=root] queued new schema change job for table 53, mutation 0
I200331 20:03:01.731859 163089 sql/sqlbase/structured.go:1554  [n1,client=127.0.0.1:60110,hostssl,user=root] publish: descID=53 (bank) version=2 mtime=1970-01-01 00:00:00 +0000 UTC
I200331 20:03:01.988686 163089 sql/event_log.go:132  [n1,client=127.0.0.1:60110,hostssl,user=root] Event: "drop_table", target: 53, info: {TableName:data.public.bank Statement:DROP TABLE data.public.bank User:root CascadeDroppedViews:[]}
I200331 20:03:01.989173 163089 jobs/registry.go:285  [n1,client=127.0.0.1:60110,hostssl,user=root] scheduled jobs [542685861200003073]
I200331 20:03:02.442760 165438 kv/kvserver/replica_command.go:663  [n1,merge,s1,r33/1:/Table/53{-/1/1}] initiating a merge of r37:/Table/53/1/{1-2} [(n1,s1):1, (n3,s3):2, (n2,s2):3, next=4, gen=9] into this range (lhs+rhs has (size=610 B+610 B=1.2 KiB qps=0.00+76.44=76.44qps) below threshold (size=128 MiB, qps=1250.00))
I200331 20:03:02.454763 157895 server/status/runtime.go:498  [n1] runtime stats: 1.3 GiB RSS, 819 goroutines, 72 MiB/99 MiB/114 MiB GO alloc/idle/total, 102 MiB/153 MiB CGO alloc/total, 1108.1 CGO/sec, 174.4/14.6 %(u/s)time, 0.6 %gc (10x), 3.9 MiB/3.9 MiB (r/w)net
I200331 20:03:02.626628 157976 jobs/jobs.go:743  job 542685861200003073: adopting
I200331 20:03:02.737945 165616 kv/kvserver/replica_command.go:663  [n3,merge,s3,r35/2:/Table/53/1/{3-4}] initiating a merge of r63:/Table/53/1/{4-5} [(n1,s1):1, (n3,s3):2, (n2,s2):3, next=4, gen=8] into this range (lhs+rhs has (size=973 B+852 B=1.8 KiB qps=2.70+3.52=6.22qps) below threshold (size=128 MiB, qps=1250.00))
I200331 20:03:02.748680 165528 kv/kvserver/replica_command.go:663  [n2,merge,s2,r34/3:/Table/53/1/{5-6}] initiating a merge of r46:/Table/53/1/{6-7} [(n1,s1):1, (n3,s3):2, (n2,s2):3, next=4, gen=9] into this range (lhs+rhs has (size=852 B+731 B=1.5 KiB qps=0.94+2.95=3.89qps) below threshold (size=128 MiB, qps=1250.00))
I200331 20:03:03.098982 157835 kv/kvserver/store_remove_replica.go:122  [n1,s1,r33/1:/Table/53{-/1/1}] removing replica r37/1
I200331 20:03:03.114367 159307 kv/kvserver/store_remove_replica.go:122  [n2,s2,r33/3:/Table/53{-/1/1}] removing replica r37/3
I200331 20:03:03.122868 159644 kv/kvserver/store_remove_replica.go:122  [n3,s3,r33/2:/Table/53{-/1/1}] removing replica r37/2
I200331 20:03:03.174329 157976 jobs/registry.go:1178  job 542685861200003073: resuming execution
I200331 20:03:03.557448 165794 kv/kvserver/replica_command.go:663  [n1,merge,s1,r45/1:/Table/53/1/{7-8}] initiating a merge of r43:/Table/53/1/{8-9} [(n1,s1):1, (n3,s3):2, (n2,s2):3, next=4, gen=8] into this range (lhs+rhs has (size=489 B+852 B=1.3 KiB qps=3.13+3.25=6.38qps) below threshold (size=128 MiB, qps=1250.00))
W200331 20:03:03.568311 165823 kv/kvserver/intentresolver/intent_resolver.go:428  failed to push during intent resolution: failed to push meta={id=48ccbb1d key=/Local/Range/Table/53/1/3/RangeDescriptor pri=0.02099490 epo=0 ts=1585684982.569595631,0 min=1585684982.569595631,0 seq=8} lock=true stat=PENDING rts=0,0 wto=false max=0,0
I200331 20:03:03.599538 165908 jobs/registry.go:797  [n1] job 542685861200003073: stepping through state running with error <nil>
I200331 20:03:03.627297 165908 sql/schema_changer.go:445  [n1,job=542685861200003073,scExec] schema change on bank (53 v2) mutation 0 starting execution...
I200331 20:03:03.739026 159538 kv/kvserver/store_remove_replica.go:122  [n3,s3,r35/2:/Table/53/1/{3-4}] removing replica r63/2
I200331 20:03:03.755164 165908 sql/lease.go:337  [n1,job=542685861200003073,scExec] waiting for 3 leases to expire: desc=[{bank 53 1}]
I200331 20:03:03.755831 157819 kv/kvserver/store_remove_replica.go:122  [n1,s1,r35/1:/Table/53/1/{3-4}] removing replica r63/1
I200331 20:03:03.763657 159302 kv/kvserver/store_remove_replica.go:122  [n2,s2,r35/3:/Table/53/1/{3-4}] removing replica r63/3
I200331 20:03:03.947343 165908 sql/lease.go:337  [n1,job=542685861200003073,scExec] waiting for 2 leases to expire: desc=[{bank 53 1}]
I200331 20:03:04.022084 159312 kv/kvserver/store_remove_replica.go:122  [n2,s2,r34/3:/Table/53/1/{5-6}] removing replica r46/3
I200331 20:03:04.044853 159612 kv/kvserver/store_remove_replica.go:122  [n3,s3,r34/2:/Table/53/1/{5-6}] removing replica r46/2
I200331 20:03:04.145459 157836 kv/kvserver/store_remove_replica.go:122  [n1,s1,r34/1:/Table/53/1/{5-6}] removing replica r46/1
I200331 20:03:04.170863 165994 kv/kvserver/replica_command.go:663  [n2,merge,s2,r36/3:/Table/53/1/{2-3}] initiating a merge of r35:/Table/53/1/{3-5} [(n1,s1):1, (n3,s3):2, (n2,s2):3, next=4, gen=9] into this range (lhs+rhs has (size=852 B+1.8 KiB=2.6 KiB qps=0.00+2.07=2.07qps) below threshold (size=128 MiB, qps=1250.00))
I200331 20:03:04.376231 157847 kv/kvserver/store_remove_replica.go:122  [n1,s1,r45/1:/Table/53/1/{7-8}] removing replica r43/1
I200331 20:03:04.405159 159599 kv/kvserver/store_remove_replica.go:122  [n3,s3,r45/2:/Table/53/1/{7-8}] removing replica r43/2
I200331 20:03:04.435217 159276 kv/kvserver/store_remove_replica.go:122  [n2,s2,r45/3:/Table/53/1/{7-8}] removing replica r43/3
I200331 20:03:04.828461 165908 sql/lease.go:337  [n1,job=542685861200003073,scExec] waiting for 1 leases to expire: desc=[{bank 53 1}]
I200331 20:03:04.866506 159330 kv/kvserver/store_remove_replica.go:122  [n2,s2,r36/3:/Table/53/1/{2-3}] removing replica r35/3
I200331 20:03:04.892906 157854 kv/kvserver/store_remove_replica.go:122  [n1,s1,r36/1:/Table/53/1/{2-3}] removing replica r35/1
I200331 20:03:04.909432 159443 kv/kvserver/store_remove_replica.go:122  [n3,s3,r36/2:/Table/53/1/{2-3}] removing replica r35/2
I200331 20:03:05.315011 166109 kv/kvserver/replica_command.go:663  [n1,merge,s1,r33/1:/Table/53{-/1/2}] initiating a merge of r36:/Table/53/1/{2-5} [(n1,s1):1, (n3,s3):2, (n2,s2):3, next=4, gen=10] into this range (lhs+rhs has (size=1.2 KiB+2.6 KiB=3.8 KiB qps=21.48+0.00=21.48qps) below threshold (size=128 MiB, qps=1250.00))
I200331 20:03:05.348330 166230 kv/kvserver/replica_command.go:663  [n2,merge,s2,r36/3:/Table/53/1/{2-5}] initiating a merge of r34:/Table/53/1/{5-7} [(n1,s1):1, (n3,s3):2, (n2,s2):3, next=4, gen=10] into this range (lhs+rhs has (size=2.6 KiB+1.5 KiB=4.2 KiB qps=5.95+3.14=9.09qps) below threshold (size=128 MiB, qps=1250.00))
I200331 20:03:05.632672 165908 sql/sqlbase/structured.go:1554  [n1,job=542685861200003073,scExec] publish: descID=53 (bank) version=3 mtime=1970-01-01 00:00:00 +0000 UTC
E200331 20:03:06.033582 163491 sql/conn_executor.go:796  [n3,client=127.0.0.1:44182,hostssl,user=root] a SQL panic has occurred while executing "UPDATE data.bank SET payload = $1 WHERE id = $2": event RetriableErr{CanAutoRetry:false, IsCommit:false} inappropriate in current state NoTxn{}
E200331 20:03:06.035217 163491 util/log/crash_reporting.go:208  [n3,client=127.0.0.1:44182,hostssl,user=root] a panic has occurred!

```

</p></details>

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestBackupRestoreWithConcurrentWrites PKG=./pkg/ccl/backupccl TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestBackupRestoreWithConcurrentWrites.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(ccl/backupccl).TestBackupRestoreWithConcurrentWrites failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1840952&tab=buildLog) on [master@bb9527175e49e3ba165c343948fd6680d2039d16](https://github.com/cockroachdb/cockroach/commits/bb9527175e49e3ba165c343948fd6680d2039d16):

Fatal error:
```
panic: event RetriableErr{CanAutoRetry:false, IsCommit:false} inappropriate in current state NoTxn{} [recovered]
	panic: panic while executing 1 statements: UPDATE _._ SET _ = $1 WHERE _ = $2; caused by event RetriableErr{CanAutoRetry:false, IsCommit:false} inappropriate in current state NoTxn{}

```

Stack:
```
goroutine 168368 [running]:
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).closeWrapper(0xc001a65000, 0x6ea6a60, 0xc005560d00, 0x5c50140, 0xc0003153c0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:810 +0x4e1
github.com/cockroachdb/cockroach/pkg/sql.(*Server).ServeConn.func1(0xc001a65000, 0x6ea6a60, 0xc005560d00)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:474 +0x6f
panic(0x5c50140, 0xc0003153c0)
	/usr/local/go/src/runtime/panic.go:679 +0x1b2
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).txnStateTransitionsApplyWrapper(0xc001a65000, 0x6e29320, 0xc000315380, 0x5c4eb60, 0xc004848c30, 0x7fa4f4052208, 0xc00066e680, 0x267, 0x0, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:2085 +0x16b5
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execCmd(0xc001a65000, 0x6ea6a60, 0xc005560d00, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1497 +0xa77
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).run(0xc001a65000, 0x6ea6a60, 0xc005560d00, 0xc003fa1a10, 0x5400, 0x15000, 0xc003fa1ab0, 0xc004f50c30, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1297 +0x484
github.com/cockroachdb/cockroach/pkg/sql.(*Server).ServeConn(0xc00227e580, 0x6ea6a60, 0xc005560d00, 0xc001a65000, 0x5400, 0x15000, 0xc003fa1ab0, 0xc004f50c30, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:476 +0x132
github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*conn).processCommandsAsync.func1(0xc0028ea8dd, 0xc0027c2980, 0x6ea6a60, 0xc005560d00, 0xc004f50c30, 0xc00227e580, 0xc00066e000, 0x6ecada0, 0xc005be2f00, 0xc002a4b0e0, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn.go:594 +0x36c
created by github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*conn).processCommandsAsync
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn.go:526 +0x1ea

```

<details><summary>Log preceding fatal error</summary><p>

```
I200331 22:27:54.011164 162825 kv/kvserver/store_remove_replica.go:122  [n1,s1,r38/1:/Table/53/1/{1-2}] removing replica r36/1
I200331 22:27:54.055464 170461 kv/kvserver/replica_command.go:663  [n3,merge,s3,r63/2:/Table/53/1/{4-5}] initiating a merge of r34:/Table/53/1/{5-6} [(n1,s1):1, (n3,s3):2, (n2,s2):3, next=4, gen=9] into this range (lhs+rhs has (size=1.9 KiB+2.3 KiB=4.1 KiB qps=1.18+6.79=7.96qps) below threshold (size=128 MiB, qps=1250.00))
I200331 22:27:54.972990 164363 server/status/runtime.go:498  [n2] runtime stats: 1.2 GiB RSS, 805 goroutines, 60 MiB/108 MiB/119 MiB GO alloc/idle/total, 99 MiB/144 MiB CGO alloc/total, 915.1 CGO/sec, 174.2/14.3 %(u/s)time, 1.0 %gc (9x), 4.0 MiB/4.0 MiB (r/w)net
I200331 22:27:55.210145 164615 kv/kvserver/store_remove_replica.go:122  [n3,s3,r63/2:/Table/53/1/{4-5}] removing replica r34/2
I200331 22:27:55.238881 164285 kv/kvserver/store_remove_replica.go:122  [n2,s2,r63/3:/Table/53/1/{4-5}] removing replica r34/3
I200331 22:27:55.313601 162820 kv/kvserver/store_remove_replica.go:122  [n1,s1,r63/1:/Table/53/1/{4-5}] removing replica r34/1
I200331 22:27:55.436046 170591 kv/kvserver/replica_command.go:663  [n2,merge,s2,r35/3:/Table/53/1/{3-4}] initiating a merge of r63:/Table/53/1/{4-6} [(n1,s1):1, (n3,s3):2, (n2,s2):3, next=4, gen=10] into this range (lhs+rhs has (size=1.5 KiB+2.1 KiB=3.7 KiB qps=0.89+1.18=2.07qps) below threshold (size=128 MiB, qps=1250.00))
I200331 22:27:55.745323 164638 kv/kvserver/store_remove_replica.go:122  [n3,s3,r33/2:/Table/53{-/1/1}] removing replica r38/2
I200331 22:27:55.747421 162790 kv/kvserver/store_remove_replica.go:122  [n1,s1,r33/1:/Table/53{-/1/1}] removing replica r38/1
I200331 22:27:55.793777 164264 kv/kvserver/store_remove_replica.go:122  [n2,s2,r33/3:/Table/53{-/1/1}] removing replica r38/3
I200331 22:27:55.823204 164679 server/status/runtime.go:498  [n3] runtime stats: 1.2 GiB RSS, 807 goroutines, 69 MiB/103 MiB/115 MiB GO alloc/idle/total, 99 MiB/145 MiB CGO alloc/total, 994.0 CGO/sec, 175.8/13.9 %(u/s)time, 1.0 %gc (9x), 4.0 MiB/4.0 MiB (r/w)net
I200331 22:27:55.912414 170789 kv/kvserver/replica_command.go:663  [n1,merge,s1,r43/1:/Table/53/1/{8-9}] initiating a merge of r37:/{Table/53/1/9-Max} [(n1,s1):1, (n3,s3):2, (n2,s2):3, next=4, gen=8] into this range (lhs+rhs has (size=2.6 KiB+2.0 KiB=4.6 KiB qps=1.55+0.00=1.55qps) below threshold (size=128 MiB, qps=1250.00))
I200331 22:27:57.057284 164295 kv/kvserver/store_remove_replica.go:122  [n2,s2,r35/3:/Table/53/1/{3-4}] removing replica r63/3
I200331 22:27:57.109501 164616 kv/kvserver/store_remove_replica.go:122  [n3,s3,r35/2:/Table/53/1/{3-4}] removing replica r63/2
I200331 22:27:57.113415 167207 sql/table.go:942  [n1,client=127.0.0.1:35394,hostssl,user=root] queued new schema change job for table 53, mutation 0
I200331 22:27:57.113678 167207 sql/sqlbase/structured.go:1554  [n1,client=127.0.0.1:35394,hostssl,user=root] publish: descID=53 (bank) version=2 mtime=1970-01-01 00:00:00 +0000 UTC
I200331 22:27:57.125835 162808 kv/kvserver/store_remove_replica.go:122  [n1,s1,r35/1:/Table/53/1/{3-4}] removing replica r63/1
I200331 22:27:57.517249 162850 kv/kvserver/store_remove_replica.go:122  [n1,s1,r43/1:/Table/53/1/{8-9}] removing replica r37/1
I200331 22:27:57.539619 164593 kv/kvserver/store_remove_replica.go:122  [n3,s3,r43/2:/Table/53/1/{8-9}] removing replica r37/2
I200331 22:27:57.591442 170948 kv/kvserver/replica_command.go:663  [n2,merge,s2,r45/3:/Table/53/1/{6-8}] initiating a merge of r43:/{Table/53/1/8-Max} [(n1,s1):1, (n3,s3):2, (n2,s2):3, next=4, gen=9] into this range (lhs+rhs has (size=6.5 KiB+2.7 KiB=9.2 KiB qps=8.03+4.13=12.16qps) below threshold (size=128 MiB, qps=1250.00))
I200331 22:27:57.657614 164323 kv/kvserver/store_remove_replica.go:122  [n2,s2,r43/3:/Table/53/1/{8-9}] removing replica r37/3
I200331 22:27:57.895509 171018 kv/kvserver/replica_command.go:663  [n1,merge,s1,r33/1:/Table/53{-/1/3}] initiating a merge of r35:/Table/53/1/{3-6} [(n1,s1):1, (n3,s3):2, (n2,s2):3, next=4, gen=11] into this range (lhs+rhs has (size=8.6 KiB+6.8 KiB=15 KiB qps=7.72+9.55=17.27qps) below threshold (size=128 MiB, qps=1250.00))
I200331 22:27:58.987843 164305 kv/kvserver/store_remove_replica.go:122  [n2,s2,r45/3:/Table/53/1/{6-8}] removing replica r43/3
I200331 22:27:59.006864 162806 kv/kvserver/store_remove_replica.go:122  [n1,s1,r45/1:/Table/53/1/{6-8}] removing replica r43/1
I200331 22:27:59.025077 164603 kv/kvserver/store_remove_replica.go:122  [n3,s3,r45/2:/Table/53/1/{6-8}] removing replica r43/2
I200331 22:27:59.085422 162802 kv/kvserver/store_remove_replica.go:122  [n1,s1,r33/1:/Table/53{-/1/3}] removing replica r35/1
I200331 22:27:59.130964 164272 kv/kvserver/store_remove_replica.go:122  [n2,s2,r33/3:/Table/53{-/1/3}] removing replica r35/3
I200331 22:27:59.193627 164581 kv/kvserver/store_remove_replica.go:122  [n3,s3,r33/2:/Table/53{-/1/3}] removing replica r35/2
I200331 22:27:59.669512 167207 sql/table.go:942  [n1,client=127.0.0.1:35394,hostssl,user=root] queued new schema change job for table 53, mutation 0
I200331 22:27:59.669880 167207 sql/sqlbase/structured.go:1554  [n1,client=127.0.0.1:35394,hostssl,user=root] publish: descID=53 (bank) version=2 mtime=1970-01-01 00:00:00 +0000 UTC
I200331 22:28:00.575365 162904 server/status/runtime.go:498  [n1] runtime stats: 1.2 GiB RSS, 792 goroutines, 44 MiB/125 MiB/120 MiB GO alloc/idle/total, 101 MiB/148 MiB CGO alloc/total, 1062.8 CGO/sec, 172.3/13.9 %(u/s)time, 0.8 %gc (9x), 4.2 MiB/4.2 MiB (r/w)net
I200331 22:28:01.058492 167207 sql/table.go:942  [n1,client=127.0.0.1:35394,hostssl,user=root] queued new schema change job for table 53, mutation 0
I200331 22:28:01.058757 167207 sql/sqlbase/structured.go:1554  [n1,client=127.0.0.1:35394,hostssl,user=root] publish: descID=53 (bank) version=2 mtime=1970-01-01 00:00:00 +0000 UTC
I200331 22:28:01.379680 171466 kv/kvserver/replica_command.go:663  [n1,merge,s1,r33/1:/Table/53{-/1/6}] initiating a merge of r45:/{Table/53/1/6-Max} [(n1,s1):1, (n3,s3):2, (n2,s2):3, next=4, gen=11] into this range (lhs+rhs has (size=19 KiB+15 KiB=34 KiB qps=56.64+26.27=82.92qps) below threshold (size=128 MiB, qps=1250.00))
I200331 22:28:01.927517 167207 sql/event_log.go:132  [n1,client=127.0.0.1:35394,hostssl,user=root] Event: "drop_table", target: 53, info: {TableName:data.public.bank Statement:DROP TABLE data.public.bank User:root CascadeDroppedViews:[]}
I200331 22:28:01.927807 167207 sql/event_log.go:132  [n1,client=127.0.0.1:35394,hostssl,user=root] Event: "drop_table", target: 53, info: {TableName:data.public.bank Statement:DROP TABLE data.public.bank User:root CascadeDroppedViews:[]}
I200331 22:28:01.927917 167207 sql/event_log.go:132  [n1,client=127.0.0.1:35394,hostssl,user=root] Event: "drop_table", target: 53, info: {TableName:data.public.bank Statement:DROP TABLE data.public.bank User:root CascadeDroppedViews:[]}
I200331 22:28:01.928062 167207 sql/event_log.go:132  [n1,client=127.0.0.1:35394,hostssl,user=root] Event: "drop_table", target: 53, info: {TableName:data.public.bank Statement:DROP TABLE data.public.bank User:root CascadeDroppedViews:[]}
I200331 22:28:01.928170 167207 sql/event_log.go:132  [n1,client=127.0.0.1:35394,hostssl,user=root] Event: "drop_table", target: 53, info: {TableName:data.public.bank Statement:DROP TABLE data.public.bank User:root CascadeDroppedViews:[]}
I200331 22:28:01.941427 167207 jobs/registry.go:285  [n1,client=127.0.0.1:35394,hostssl,user=root] scheduled jobs [542714366533599233]
I200331 22:28:02.379693 162835 kv/kvserver/store_remove_replica.go:122  [n1,s1,r33/1:/Table/53{-/1/6}] removing replica r45/1
I200331 22:28:02.401172 164305 kv/kvserver/store_remove_replica.go:122  [n2,s2,r33/3:/Table/53{-/1/6}] removing replica r45/3
I200331 22:28:02.422513 164580 kv/kvserver/store_remove_replica.go:122  [n3,s3,r33/2:/Table/53{-/1/6}] removing replica r45/2
I200331 22:28:02.896690 162899 jobs/jobs.go:743  job 542714366533599233: adopting
I200331 22:28:03.635913 162899 jobs/registry.go:1178  job 542714366533599233: resuming execution
I200331 22:28:03.775516 172074 jobs/registry.go:797  [n1] job 542714366533599233: stepping through state running with error <nil>
I200331 22:28:03.796406 172074 sql/schema_changer.go:445  [n1,job=542714366533599233,scExec] schema change on bank (53 v2) mutation 0 starting execution...
I200331 22:28:04.041416 172074 sql/sqlbase/structured.go:1554  [n1,job=542714366533599233,scExec] publish: descID=53 (bank) version=3 mtime=1970-01-01 00:00:00 +0000 UTC
E200331 22:28:04.596246 168368 sql/conn_executor.go:796  [n1,client=127.0.0.1:35408,hostssl,user=root] a SQL panic has occurred while executing "UPDATE data.bank SET payload = $1 WHERE id = $2": event RetriableErr{CanAutoRetry:false, IsCommit:false} inappropriate in current state NoTxn{}
E200331 22:28:04.597359 168368 util/log/crash_reporting.go:208  [n1,client=127.0.0.1:35408,hostssl,user=root] a panic has occurred!

```

</p></details>

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestBackupRestoreWithConcurrentWrites PKG=./pkg/ccl/backupccl TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestBackupRestoreWithConcurrentWrites.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(ccl/backupccl).TestBackupRestoreWithConcurrentWrites failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1841364&tab=buildLog) on [master@1aa26302f721f29d793d7153b24ddf97eb5e68eb](https://github.com/cockroachdb/cockroach/commits/1aa26302f721f29d793d7153b24ddf97eb5e68eb):

Fatal error:
```
panic: event RetriableErr{CanAutoRetry:false, IsCommit:false} inappropriate in current state NoTxn{} [recovered]
	panic: panic while executing 1 statements: UPDATE _._ SET _ = $1 WHERE _ = $2; caused by event RetriableErr{CanAutoRetry:false, IsCommit:false} inappropriate in current state NoTxn{}

```

Stack:
```
goroutine 166398 [running]:
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).closeWrapper(0xc0015b0000, 0x6ea9200, 0xc001a6e680, 0x5c51f80, 0xc000fdec80)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:810 +0x4e1
github.com/cockroachdb/cockroach/pkg/sql.(*Server).ServeConn.func1(0xc0015b0000, 0x6ea9200, 0xc001a6e680)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:474 +0x6f
panic(0x5c51f80, 0xc000fdec80)
	/usr/local/go/src/runtime/panic.go:679 +0x1b2
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).txnStateTransitionsApplyWrapper(0xc0015b0000, 0x6e2ba80, 0xc000fdec60, 0x5c509a0, 0xc003ee5aa0, 0x7f16e4fc3ae8, 0xc002330680, 0x148, 0x0, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:2085 +0x16b5
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execCmd(0xc0015b0000, 0x6ea9200, 0xc001a6e680, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1497 +0xa77
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).run(0xc0015b0000, 0x6ea9200, 0xc001a6e680, 0xc005d2d510, 0x5400, 0x15000, 0xc005d2d5b0, 0xc0002f01b0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1297 +0x484
github.com/cockroachdb/cockroach/pkg/sql.(*Server).ServeConn(0xc0057ff600, 0x6ea9200, 0xc001a6e680, 0xc0015b0000, 0x5400, 0x15000, 0xc005d2d5b0, 0xc0002f01b0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:476 +0x132
github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*conn).processCommandsAsync.func1(0xc001cc427d, 0xc004cb93e0, 0x6ea9200, 0xc001a6e680, 0xc0002f01b0, 0xc0057ff600, 0xc002330000, 0x6ecd540, 0xc0047d68d0, 0xc00075b8c0, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn.go:594 +0x36c
created by github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*conn).processCommandsAsync
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn.go:526 +0x1ea

```

<details><summary>Log preceding fatal error</summary><p>

```
I200401 03:04:53.825786 167422 jobs/registry.go:797  [n1] job 542768792172822529: stepping through state succeeded with error <nil>
I200401 03:04:54.369650 162958 server/status/runtime.go:498  [n3] runtime stats: 1.2 GiB RSS, 772 goroutines, 68 MiB/103 MiB/114 MiB GO alloc/idle/total, 94 MiB/139 MiB CGO alloc/total, 2731.9 CGO/sec, 200.1/17.9 %(u/s)time, 0.9 %gc (10x), 3.9 MiB/3.9 MiB (r/w)net
I200401 03:04:54.439091 167422 jobs/registry.go:952  [n1] job 542768792172822529: status succeeded after adoption finished
I200401 03:04:56.806980 168138 kv/kvserver/replica_command.go:663  [n2,merge,s2,r37/2:/Table/53/1/{2-3}] initiating a merge of r35:/Table/53/1/{3-4} [(n1,s1):1, (n2,s2):2, (n3,s3):3, next=4, gen=8] into this range (lhs+rhs has (size=1.9 KiB+1.5 KiB=3.4 KiB qps=10.54+5.87=16.40qps) below threshold (size=128 MiB, qps=1250.00))
I200401 03:04:57.862279 162532 kv/kvserver/store_remove_replica.go:122  [n2,s2,r37/2:/Table/53/1/{2-3}] removing replica r35/2
I200401 03:04:57.875292 162913 kv/kvserver/store_remove_replica.go:122  [n3,s3,r37/3:/Table/53/1/{2-3}] removing replica r35/3
I200401 03:04:57.882564 161111 kv/kvserver/store_remove_replica.go:122  [n1,s1,r37/1:/Table/53/1/{2-3}] removing replica r35/1
I200401 03:04:58.068078 168364 kv/kvserver/replica_command.go:663  [n2,merge,s2,r38/2:/Table/53/1/{7-8}] initiating a merge of r36:/Table/53/1/{8-9} [(n1,s1):1, (n2,s2):2, (n3,s3):3, next=4, gen=8] into this range (lhs+rhs has (size=1.4 KiB+2.6 KiB=4.0 KiB qps=2.54+15.44=17.98qps) below threshold (size=128 MiB, qps=1250.00))
I200401 03:04:58.516574 166497 sql/table.go:942  [n1,client=127.0.0.1:36704,hostssl,user=root] queued new schema change job for table 53, mutation 0
I200401 03:04:58.516805 166497 sql/sqlbase/structured.go:1554  [n1,client=127.0.0.1:36704,hostssl,user=root] publish: descID=53 (bank) version=2 mtime=1970-01-01 00:00:00 +0000 UTC
I200401 03:04:58.577952 162590 kv/kvserver/store_remove_replica.go:122  [n2,s2,r38/2:/Table/53/1/{7-8}] removing replica r36/2
I200401 03:04:58.578291 161146 kv/kvserver/store_remove_replica.go:122  [n1,s1,r38/1:/Table/53/1/{7-8}] removing replica r36/1
I200401 03:04:58.594850 162904 kv/kvserver/store_remove_replica.go:122  [n3,s3,r38/3:/Table/53/1/{7-8}] removing replica r36/3
I200401 03:04:58.686135 168419 kv/kvserver/replica_command.go:663  [n2,merge,s2,r39/2:/Table/53/1/{1-2}] initiating a merge of r37:/Table/53/1/{2-4} [(n1,s1):1, (n2,s2):2, (n3,s3):3, next=4, gen=9] into this range (lhs+rhs has (size=2.3 KiB+4.9 KiB=7.1 KiB qps=6.52+17.71=24.23qps) below threshold (size=128 MiB, qps=1250.00))
I200401 03:04:58.833487 168342 kv/kvserver/replica_command.go:663  [n3,merge,s3,r40/3:/Table/53/1/{6-7}] initiating a merge of r38:/Table/53/1/{7-9} [(n1,s1):1, (n2,s2):2, (n3,s3):3, next=4, gen=9] into this range (lhs+rhs has (size=1.8 KiB+1.4 KiB=3.2 KiB qps=4.14+2.54=6.68qps) below threshold (size=128 MiB, qps=1250.00))
I200401 03:04:59.055697 161194 server/status/runtime.go:498  [n1] runtime stats: 1.2 GiB RSS, 801 goroutines, 52 MiB/120 MiB/116 MiB GO alloc/idle/total, 98 MiB/143 MiB CGO alloc/total, 1877.0 CGO/sec, 201.7/16.6 %(u/s)time, 1.3 %gc (11x), 4.1 MiB/4.1 MiB (r/w)net
I200401 03:04:59.263956 166497 sql/event_log.go:132  [n1,client=127.0.0.1:36704,hostssl,user=root] Event: "drop_table", target: 53, info: {TableName:data.public.bank Statement:DROP TABLE data.public.bank User:root CascadeDroppedViews:[]}
I200401 03:04:59.275775 166497 jobs/registry.go:285  [n1,client=127.0.0.1:36704,hostssl,user=root] scheduled jobs [542768818944704513]
I200401 03:04:59.613292 162915 kv/kvserver/store_remove_replica.go:122  [n3,s3,r40/3:/Table/53/1/{6-7}] removing replica r38/3
I200401 03:04:59.722387 162598 kv/kvserver/store_remove_replica.go:122  [n2,s2,r40/2:/Table/53/1/{6-7}] removing replica r38/2
I200401 03:04:59.732797 162556 kv/kvserver/store_remove_replica.go:122  [n2,s2,r39/2:/Table/53/1/{1-2}] removing replica r37/2
I200401 03:04:59.739169 161117 kv/kvserver/store_remove_replica.go:122  [n1,s1,r40/1:/Table/53/1/{6-7}] removing replica r38/1
I200401 03:04:59.752092 161147 kv/kvserver/store_remove_replica.go:122  [n1,s1,r39/1:/Table/53/1/{1-2}] removing replica r37/1
I200401 03:04:59.785550 162890 kv/kvserver/store_remove_replica.go:122  [n3,s3,r39/3:/Table/53/1/{1-2}] removing replica r37/3
I200401 03:04:59.940864 168775 kv/kvserver/replica_command.go:663  [n1,merge,s1,r33/1:/Table/53{-/1/1}] initiating a merge of r39:/Table/53/1/{1-4} [(n1,s1):1, (n2,s2):2, (n3,s3):3, next=4, gen=10] into this range (lhs+rhs has (size=2.1 KiB+2.4 KiB=4.5 KiB qps=5.80+8.93=14.73qps) below threshold (size=128 MiB, qps=1250.00))
I200401 03:05:00.191601 168700 kv/kvserver/replica_command.go:663  [n3,merge,s3,r53/3:/Table/53/1/{4-5}] initiating a merge of r34:/Table/53/1/{5-6} [(n1,s1):1, (n2,s2):2, (n3,s3):3, next=4, gen=9] into this range (lhs+rhs has (size=1.5 KiB+2.8 KiB=4.4 KiB qps=2.81+0.00=2.81qps) below threshold (size=128 MiB, qps=1250.00))
I200401 03:05:00.298941 168879 kv/kvserver/replica_command.go:663  [n2,merge,s2,r39/2:/Table/53/1/{1-4}] initiating a merge of r53:/Table/53/1/{4-5} [(n1,s1):1, (n2,s2):2, (n3,s3):3, next=4, gen=8] into this range (lhs+rhs has (size=7.3 KiB+1.5 KiB=8.9 KiB qps=8.93+2.81=11.74qps) below threshold (size=128 MiB, qps=1250.00))
I200401 03:05:00.696445 161290 jobs/jobs.go:743  job 542768818944704513: adopting
I200401 03:05:02.173927 168700 kv/kvserver/replica_command.go:663  [n3,merge,s3,r53/3:/Table/53/1/{4-5}] initiating a merge of r34:/Table/53/1/{5-6} [(n1,s1):1, (n2,s2):2, (n3,s3):3, next=4, gen=9] into this range (lhs+rhs has (size=1.5 KiB+2.8 KiB=4.4 KiB qps=2.81+0.00=2.81qps) below threshold (size=128 MiB, qps=1250.00))
I200401 03:05:02.510018 161290 jobs/registry.go:1178  job 542768818944704513: resuming execution
I200401 03:05:02.600497 168775 kv/kvserver/replica_command.go:663  [n1,merge,s1,r33/1:/Table/53{-/1/1}] initiating a merge of r39:/Table/53/1/{1-4} [(n1,s1):1, (n2,s2):2, (n3,s3):3, next=4, gen=10] into this range (lhs+rhs has (size=2.1 KiB+2.4 KiB=4.5 KiB qps=5.80+8.93=14.73qps) below threshold (size=128 MiB, qps=1250.00))
I200401 03:05:03.096288 169280 jobs/registry.go:797  [n1] job 542768818944704513: stepping through state running with error <nil>
I200401 03:05:03.287406 169280 sql/schema_changer.go:445  [n1,job=542768818944704513,scExec] schema change on bank (53 v2) mutation 0 starting execution...
I200401 03:05:03.440824 162629 server/status/runtime.go:498  [n2] runtime stats: 1.2 GiB RSS, 801 goroutines, 75 MiB/99 MiB/116 MiB GO alloc/idle/total, 99 MiB/148 MiB CGO alloc/total, 1002.4 CGO/sec, 188.5/15.1 %(u/s)time, 1.4 %gc (11x), 4.1 MiB/4.1 MiB (r/w)net
I200401 03:05:03.515471 162898 kv/kvserver/store_remove_replica.go:122  [n3,s3,r53/3:/Table/53/1/{4-5}] removing replica r34/3
I200401 03:05:03.537189 162557 kv/kvserver/store_remove_replica.go:122  [n2,s2,r53/2:/Table/53/1/{4-5}] removing replica r34/2
I200401 03:05:03.551158 161113 kv/kvserver/store_remove_replica.go:122  [n1,s1,r53/1:/Table/53/1/{4-5}] removing replica r34/1
I200401 03:05:03.691444 168879 kv/kvserver/replica_command.go:663  [n2,merge,s2,r39/2:/Table/53/1/{1-4}] initiating a merge of r53:/Table/53/1/{4-6} [(n1,s1):1, (n2,s2):2, (n3,s3):3, next=4, gen=10] into this range (lhs+rhs has (size=7.3 KiB+1.5 KiB=8.9 KiB qps=8.93+2.81=11.74qps) below threshold (size=128 MiB, qps=1250.00))
I200401 03:05:03.773256 169450 kv/kvserver/replica_command.go:663  [n3,merge,s3,r40/3:/Table/53/1/{6-9}] initiating a merge of r54:/{Table/53/1/9-Max} [(n1,s1):1, (n2,s2):2, (n3,s3):3, next=4, gen=8] into this range (lhs+rhs has (size=6.8 KiB+2.5 KiB=9.2 KiB qps=0.00+0.00=0.00qps) below threshold (size=128 MiB, qps=1250.00))
I200401 03:05:03.876846 161108 kv/kvserver/store_remove_replica.go:122  [n1,s1,r33/1:/Table/53{-/1/1}] removing replica r39/1
I200401 03:05:03.902835 169280 sql/sqlbase/structured.go:1554  [n1,job=542768818944704513,scExec] publish: descID=53 (bank) version=3 mtime=1970-01-01 00:00:00 +0000 UTC
I200401 03:05:03.959759 162593 kv/kvserver/store_remove_replica.go:122  [n2,s2,r33/2:/Table/53{-/1/1}] removing replica r39/2
I200401 03:05:03.966919 162918 kv/kvserver/store_remove_replica.go:122  [n3,s3,r33/3:/Table/53{-/1/1}] removing replica r39/3
I200401 03:05:04.361203 169526 kv/kvserver/replica_command.go:663  [n1,merge,s1,r33/1:/Table/53{-/1/4}] initiating a merge of r53:/Table/53/1/{4-6} [(n1,s1):1, (n2,s2):2, (n3,s3):3, next=4, gen=10] into this range (lhs+rhs has (size=9.6 KiB+4.4 KiB=14 KiB qps=5.88+6.85=12.73qps) below threshold (size=128 MiB, qps=1250.00))
I200401 03:05:04.438108 162958 server/status/runtime.go:498  [n3] runtime stats: 1.2 GiB RSS, 811 goroutines, 78 MiB/95 MiB/118 MiB GO alloc/idle/total, 99 MiB/147 MiB CGO alloc/total, 1041.6 CGO/sec, 186.1/15.6 %(u/s)time, 1.5 %gc (11x), 4.0 MiB/4.0 MiB (r/w)net
I200401 03:05:04.634844 162923 kv/kvserver/store_remove_replica.go:122  [n3,s3,r40/3:/Table/53/1/{6-9}] removing replica r54/3
I200401 03:05:04.695612 162608 kv/kvserver/store_remove_replica.go:122  [n2,s2,r40/2:/Table/53/1/{6-9}] removing replica r54/2
I200401 03:05:04.716095 161117 kv/kvserver/store_remove_replica.go:122  [n1,s1,r40/1:/Table/53/1/{6-9}] removing replica r54/1
E200401 03:05:04.858349 166398 sql/conn_executor.go:796  [n1,client=127.0.0.1:36700,hostssl,user=root] a SQL panic has occurred while executing "UPDATE data.bank SET payload = $1 WHERE id = $2": event RetriableErr{CanAutoRetry:false, IsCommit:false} inappropriate in current state NoTxn{}
E200401 03:05:04.859203 166398 util/log/crash_reporting.go:208  [n1,client=127.0.0.1:36700,hostssl,user=root] a panic has occurred!

```

</p></details>

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestBackupRestoreWithConcurrentWrites PKG=./pkg/ccl/backupccl TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```

Related:
- #46827 ccl/backupccl: TestBackupRestoreWithConcurrentWrites failed  [C-test-failure](https://api.github.com/repos/cockroachdb/cockroach/labels/C-test-failure) [O-robot](https://api.github.com/repos/cockroachdb/cockroach/labels/O-robot) [branch-release-20.1](https://api.github.com/repos/cockroachdb/cockroach/labels/branch-release-20.1)

[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestBackupRestoreWithConcurrentWrites.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(ccl/backupccl).TestBackupRestoreWithConcurrentWrites failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1841409&tab=buildLog) on [master@9c590593565908c97b81fe697a6f31414807daf8](https://github.com/cockroachdb/cockroach/commits/9c590593565908c97b81fe697a6f31414807daf8):

Fatal error:
```
panic: event RetriableErr{CanAutoRetry:false, IsCommit:false} inappropriate in current state NoTxn{} [recovered]
	panic: panic while executing 1 statements: UPDATE _._ SET _ = $1 WHERE _ = $2; caused by event RetriableErr{CanAutoRetry:false, IsCommit:false} inappropriate in current state NoTxn{}

```

Stack:
```
goroutine 173758 [running]:
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).closeWrapper(0xc000f88000, 0x6ea9c80, 0xc001745c80, 0x5c52620, 0xc0025e4940)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:810 +0x4e1
github.com/cockroachdb/cockroach/pkg/sql.(*Server).ServeConn.func1(0xc000f88000, 0x6ea9c80, 0xc001745c80)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:474 +0x6f
panic(0x5c52620, 0xc0025e4940)
	/usr/local/go/src/runtime/panic.go:679 +0x1b2
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).txnStateTransitionsApplyWrapper(0xc000f88000, 0x6e2c540, 0xc0025e4880, 0x5c51040, 0xc006531b00, 0x7fa765011600, 0xc0022f0f80, 0x2ac, 0x0, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:2085 +0x16b5
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execCmd(0xc000f88000, 0x6ea9c80, 0xc001745c80, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1497 +0xa77
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).run(0xc000f88000, 0x6ea9c80, 0xc001745c80, 0xc000f5d290, 0x5400, 0x15000, 0xc000f5d330, 0xc0018275b0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1297 +0x484
github.com/cockroachdb/cockroach/pkg/sql.(*Server).ServeConn(0xc000e19180, 0x6ea9c80, 0xc001745c80, 0xc000f88000, 0x5400, 0x15000, 0xc000f5d330, 0xc0018275b0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:476 +0x132
github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*conn).processCommandsAsync.func1(0xc00427a27d, 0xc004333f60, 0x6ea9c80, 0xc001745c80, 0xc0018275b0, 0xc000e19180, 0xc0022f0900, 0x6ecdfc0, 0xc006014450, 0xc0044bea80, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn.go:594 +0x36c
created by github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*conn).processCommandsAsync
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn.go:526 +0x1ea

```

<details><summary>Log preceding fatal error</summary><p>

```
I200401 03:51:20.940020 173144 sql/table.go:942  [n1,client=127.0.0.1:47504,hostssl,user=root] queued new schema change job for table 53, mutation 0
I200401 03:51:20.944459 173144 sql/sqlbase/structured.go:1554  [n1,client=127.0.0.1:47504,hostssl,user=root] publish: descID=53 (bank) version=2 mtime=1970-01-01 00:00:00 +0000 UTC
I200401 03:51:21.144401 169171 kv/kvserver/store_remove_replica.go:122  [n2,s2,r35/3:/Table/53/1/{3-4}] removing replica r37/3
I200401 03:51:21.145501 167730 kv/kvserver/store_remove_replica.go:122  [n1,s1,r35/1:/Table/53/1/{3-4}] removing replica r37/1
I200401 03:51:21.218545 169538 kv/kvserver/store_remove_replica.go:122  [n3,s3,r35/2:/Table/53/1/{3-4}] removing replica r37/2
I200401 03:51:21.488510 175539 kv/kvserver/replica_command.go:663  [n1,merge,s1,r43/1:/Table/53/1/{8-9}] initiating a merge of r39:/{Table/53/1/9-Max} [(n1,s1):1, (n3,s3):2, (n2,s2):3, next=4, gen=8] into this range (lhs+rhs has (size=3.2 KiB+3.0 KiB=6.2 KiB qps=3.95+0.00=3.95qps) below threshold (size=128 MiB, qps=1250.00))
I200401 03:51:21.770195 173144 sql/table.go:942  [n1,client=127.0.0.1:47504,hostssl,user=root] queued new schema change job for table 53, mutation 0
I200401 03:51:21.770615 173144 sql/sqlbase/structured.go:1554  [n1,client=127.0.0.1:47504,hostssl,user=root] publish: descID=53 (bank) version=2 mtime=1970-01-01 00:00:00 +0000 UTC
W200401 03:51:22.408547 175735 kv/kvserver/intentresolver/intent_resolver.go:428  failed to push during intent resolution: failed to push meta={id=490723bc key=/Local/Range/Table/53/1/8/RangeDescriptor pri=0.09651566 epo=0 ts=1585713081.349402571,0 min=1585713081.349402571,0 seq=8} lock=true stat=PENDING rts=0,0 wto=false max=0,0
I200401 03:51:22.629033 169173 kv/kvserver/store_remove_replica.go:122  [n2,s2,r43/3:/Table/53/1/{8-9}] removing replica r39/3
I200401 03:51:22.730118 167627 kv/kvserver/store_remove_replica.go:122  [n1,s1,r43/1:/Table/53/1/{8-9}] removing replica r39/1
I200401 03:51:22.746074 169535 kv/kvserver/store_remove_replica.go:122  [n3,s3,r43/2:/Table/53/1/{8-9}] removing replica r39/2
I200401 03:51:22.850497 173144 sql/table.go:942  [n1,client=127.0.0.1:47504,hostssl,user=root] queued new schema change job for table 53, mutation 0
I200401 03:51:22.850917 173144 sql/sqlbase/structured.go:1554  [n1,client=127.0.0.1:47504,hostssl,user=root] publish: descID=53 (bank) version=2 mtime=1970-01-01 00:00:00 +0000 UTC
I200401 03:51:23.341131 175758 kv/kvserver/replica_command.go:663  [n3,merge,s3,r45/2:/Table/53/1/{6-8}] initiating a merge of r43:/{Table/53/1/8-Max} [(n1,s1):1, (n3,s3):2, (n2,s2):3, next=4, gen=9] into this range (lhs+rhs has (size=6.3 KiB+6.6 KiB=13 KiB qps=4.68+5.37=10.05qps) below threshold (size=128 MiB, qps=1250.00))
I200401 03:51:23.735042 173144 sql/table.go:942  [n1,client=127.0.0.1:47504,hostssl,user=root] queued new schema change job for table 53, mutation 0
I200401 03:51:23.735453 173144 sql/sqlbase/structured.go:1554  [n1,client=127.0.0.1:47504,hostssl,user=root] publish: descID=53 (bank) version=2 mtime=1970-01-01 00:00:00 +0000 UTC
I200401 03:51:24.383855 169514 kv/kvserver/store_remove_replica.go:122  [n3,s3,r45/2:/Table/53/1/{6-8}] removing replica r43/2
I200401 03:51:24.409055 167733 kv/kvserver/store_remove_replica.go:122  [n1,s1,r45/1:/Table/53/1/{6-8}] removing replica r43/1
I200401 03:51:24.439988 169162 kv/kvserver/store_remove_replica.go:122  [n2,s2,r45/3:/Table/53/1/{6-8}] removing replica r43/3
I200401 03:51:24.531252 167784 server/status/runtime.go:498  [n1] runtime stats: 1.3 GiB RSS, 786 goroutines, 54 MiB/116 MiB/118 MiB GO alloc/idle/total, 102 MiB/150 MiB CGO alloc/total, 1292.2 CGO/sec, 215.2/17.6 %(u/s)time, 0.9 %gc (10x), 3.7 MiB/3.7 MiB (r/w)net
I200401 03:51:24.623661 175872 kv/kvserver/replica_command.go:663  [n1,merge,s1,r33/1:/Table/53{-/1/3}] initiating a merge of r35:/Table/53/1/{3-6} [(n1,s1):1, (n3,s3):2, (n2,s2):3, next=4, gen=11] into this range (lhs+rhs has (size=10 KiB+9.3 KiB=20 KiB qps=18.20+16.15=34.35qps) below threshold (size=128 MiB, qps=1250.00))
I200401 03:51:25.033099 167721 kv/kvserver/store_remove_replica.go:122  [n1,s1,r33/1:/Table/53{-/1/3}] removing replica r35/1
I200401 03:51:25.047552 169492 kv/kvserver/store_remove_replica.go:122  [n3,s3,r33/2:/Table/53{-/1/3}] removing replica r35/2
I200401 03:51:25.078747 169122 kv/kvserver/store_remove_replica.go:122  [n2,s2,r33/3:/Table/53{-/1/3}] removing replica r35/3
I200401 03:51:25.295258 173144 sql/table.go:942  [n1,client=127.0.0.1:47504,hostssl,user=root] queued new schema change job for table 53, mutation 0
I200401 03:51:25.295576 173144 sql/sqlbase/structured.go:1554  [n1,client=127.0.0.1:47504,hostssl,user=root] publish: descID=53 (bank) version=2 mtime=1970-01-01 00:00:00 +0000 UTC
I200401 03:51:25.998825 173144 sql/table.go:942  [n1,client=127.0.0.1:47504,hostssl,user=root] queued new schema change job for table 53, mutation 0
I200401 03:51:25.999304 173144 sql/sqlbase/structured.go:1554  [n1,client=127.0.0.1:47504,hostssl,user=root] publish: descID=53 (bank) version=2 mtime=1970-01-01 00:00:00 +0000 UTC
I200401 03:51:26.374993 173144 sql/event_log.go:132  [n1,client=127.0.0.1:47504,hostssl,user=root] Event: "drop_table", target: 53, info: {TableName:data.public.bank Statement:DROP TABLE data.public.bank User:root CascadeDroppedViews:[]}
I200401 03:51:26.386681 173144 sql/event_log.go:132  [n1,client=127.0.0.1:47504,hostssl,user=root] Event: "drop_table", target: 53, info: {TableName:data.public.bank Statement:DROP TABLE data.public.bank User:root CascadeDroppedViews:[]}
I200401 03:51:26.387249 173144 sql/event_log.go:132  [n1,client=127.0.0.1:47504,hostssl,user=root] Event: "drop_table", target: 53, info: {TableName:data.public.bank Statement:DROP TABLE data.public.bank User:root CascadeDroppedViews:[]}
I200401 03:51:26.387613 173144 sql/event_log.go:132  [n1,client=127.0.0.1:47504,hostssl,user=root] Event: "drop_table", target: 53, info: {TableName:data.public.bank Statement:DROP TABLE data.public.bank User:root CascadeDroppedViews:[]}
I200401 03:51:26.388027 173144 sql/event_log.go:132  [n1,client=127.0.0.1:47504,hostssl,user=root] Event: "drop_table", target: 53, info: {TableName:data.public.bank Statement:DROP TABLE data.public.bank User:root CascadeDroppedViews:[]}
I200401 03:51:26.388430 173144 sql/event_log.go:132  [n1,client=127.0.0.1:47504,hostssl,user=root] Event: "drop_table", target: 53, info: {TableName:data.public.bank Statement:DROP TABLE data.public.bank User:root CascadeDroppedViews:[]}
I200401 03:51:26.388796 173144 sql/event_log.go:132  [n1,client=127.0.0.1:47504,hostssl,user=root] Event: "drop_table", target: 53, info: {TableName:data.public.bank Statement:DROP TABLE data.public.bank User:root CascadeDroppedViews:[]}
I200401 03:51:26.389208 173144 sql/event_log.go:132  [n1,client=127.0.0.1:47504,hostssl,user=root] Event: "drop_table", target: 53, info: {TableName:data.public.bank Statement:DROP TABLE data.public.bank User:root CascadeDroppedViews:[]}
I200401 03:51:26.389653 173144 sql/event_log.go:132  [n1,client=127.0.0.1:47504,hostssl,user=root] Event: "drop_table", target: 53, info: {TableName:data.public.bank Statement:DROP TABLE data.public.bank User:root CascadeDroppedViews:[]}
I200401 03:51:26.392095 173144 jobs/registry.go:285  [n1,client=127.0.0.1:47504,hostssl,user=root] scheduled jobs [542777953379975169]
I200401 03:51:26.944412 167897 jobs/jobs.go:743  job 542777953379975169: adopting
I200401 03:51:26.955299 176453 kv/kvserver/replica_command.go:663  [n1,merge,s1,r33/1:/Table/53{-/1/6}] initiating a merge of r45:/{Table/53/1/6-Max} [(n1,s1):1, (n3,s3):2, (n2,s2):3, next=4, gen=11] into this range (lhs+rhs has (size=22 KiB+16 KiB=38 KiB qps=33.78+22.45=56.24qps) below threshold (size=128 MiB, qps=1250.00))
I200401 03:51:27.263999 167897 jobs/registry.go:1178  job 542777953379975169: resuming execution
I200401 03:51:27.416756 176483 jobs/registry.go:797  [n1] job 542777953379975169: stepping through state running with error <nil>
I200401 03:51:27.493352 176483 sql/schema_changer.go:445  [n1,job=542777953379975169,scExec] schema change on bank (53 v2) mutation 0 starting execution...
I200401 03:51:27.664307 176483 sql/sqlbase/structured.go:1554  [n1,job=542777953379975169,scExec] publish: descID=53 (bank) version=3 mtime=1970-01-01 00:00:00 +0000 UTC
I200401 03:51:27.858460 169495 kv/kvserver/store_remove_replica.go:122  [n3,s3,r33/2:/Table/53{-/1/6}] removing replica r45/2
I200401 03:51:27.904135 169169 kv/kvserver/store_remove_replica.go:122  [n2,s2,r33/3:/Table/53{-/1/6}] removing replica r45/3
I200401 03:51:27.917063 167724 kv/kvserver/store_remove_replica.go:122  [n1,s1,r33/1:/Table/53{-/1/6}] removing replica r45/1
E200401 03:51:28.063008 173758 sql/conn_executor.go:796  [n3,client=127.0.0.1:41252,hostssl,user=root] a SQL panic has occurred while executing "UPDATE data.bank SET payload = $1 WHERE id = $2": event RetriableErr{CanAutoRetry:false, IsCommit:false} inappropriate in current state NoTxn{}
E200401 03:51:28.064444 173758 util/log/crash_reporting.go:208  [n3,client=127.0.0.1:41252,hostssl,user=root] a panic has occurred!

```

</p></details>

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestBackupRestoreWithConcurrentWrites PKG=./pkg/ccl/backupccl TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```

Related:
- #46827 ccl/backupccl: TestBackupRestoreWithConcurrentWrites failed  [C-test-failure](https://api.github.com/repos/cockroachdb/cockroach/labels/C-test-failure) [O-robot](https://api.github.com/repos/cockroachdb/cockroach/labels/O-robot) [branch-release-20.1](https://api.github.com/repos/cockroachdb/cockroach/labels/branch-release-20.1)

[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestBackupRestoreWithConcurrentWrites.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(ccl/backupccl).TestBackupRestoreWithConcurrentWrites failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1842919&tab=buildLog) on [master@9c590593565908c97b81fe697a6f31414807daf8](https://github.com/cockroachdb/cockroach/commits/9c590593565908c97b81fe697a6f31414807daf8):

Fatal error:
```
panic: event RetriableErr{CanAutoRetry:false, IsCommit:false} inappropriate in current state NoTxn{} [recovered]
	panic: panic while executing 1 statements: UPDATE _._ SET _ = $1 WHERE _ = $2; caused by event RetriableErr{CanAutoRetry:false, IsCommit:false} inappropriate in current state NoTxn{}

```

Stack:
```
goroutine 165145 [running]:
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).closeWrapper(0xc002056000, 0x6ea9c80, 0xc0054733c0, 0x5c52620, 0xc0049d6520)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:810 +0x4e1
github.com/cockroachdb/cockroach/pkg/sql.(*Server).ServeConn.func1(0xc002056000, 0x6ea9c80, 0xc0054733c0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:474 +0x6f
panic(0x5c52620, 0xc0049d6520)
	/usr/local/go/src/runtime/panic.go:679 +0x1b2
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).txnStateTransitionsApplyWrapper(0xc002056000, 0x6e2c540, 0xc0049d6500, 0x5c51040, 0xc001d07bc0, 0x7ff7b4e527e8, 0xc001c8d380, 0x21c, 0x0, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:2085 +0x16b5
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execCmd(0xc002056000, 0x6ea9c80, 0xc0054733c0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1497 +0xa77
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).run(0xc002056000, 0x6ea9c80, 0xc0054733c0, 0xc000241010, 0x5400, 0x15000, 0xc0002410b0, 0xc0021b22b0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1297 +0x484
github.com/cockroachdb/cockroach/pkg/sql.(*Server).ServeConn(0xc0007ff700, 0x6ea9c80, 0xc0054733c0, 0xc002056000, 0x5400, 0x15000, 0xc0002410b0, 0xc0021b22b0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:476 +0x132
github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*conn).processCommandsAsync.func1(0xc004275132, 0xc0064e5ee0, 0x6ea9c80, 0xc0054733c0, 0xc0021b22b0, 0xc0007ff700, 0xc001c8cd00, 0x6ecdfc0, 0xc0007f5320, 0xc00788e6c0, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn.go:594 +0x36c
created by github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*conn).processCommandsAsync
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn.go:526 +0x1ea

```

<details><summary>Log preceding fatal error</summary><p>

```
I200401 06:20:58.175097 166659 kv/kvserver/replica_command.go:663  [n1,merge,s1,r33/1:/Table/53{-/1/1}] initiating a merge of r40:/Table/53/1/{1-4} [(n1,s1):1, (n2,s2):2, (n3,s3):3, next=4, gen=10] into this range (lhs+rhs has (size=2.3 KiB+1.9 KiB=4.1 KiB qps=12.17+6.35=18.52qps) below threshold (size=128 MiB, qps=1250.00))
I200401 06:20:58.233232 166964 kv/kvserver/replica_command.go:663  [n3,merge,s3,r39/3:/Table/53/1/{6-7}] initiating a merge of r37:/Table/53/1/{7-8} [(n1,s1):1, (n2,s2):2, (n3,s3):3, next=4, gen=8] into this range (lhs+rhs has (size=3.1 KiB+2.1 KiB=5.2 KiB qps=4.91+0.00=4.91qps) below threshold (size=128 MiB, qps=1250.00))
I200401 06:20:59.152406 164569 sql/table.go:942  [n1,client=127.0.0.1:37780,hostssl,user=root] queued new schema change job for table 53, mutation 0
I200401 06:20:59.152737 164569 sql/sqlbase/structured.go:1554  [n1,client=127.0.0.1:37780,hostssl,user=root] publish: descID=53 (bank) version=2 mtime=1970-01-01 00:00:00 +0000 UTC
I200401 06:20:59.328101 160788 kv/kvserver/store_remove_replica.go:122  [n2,s2,r35/2:/Table/53/1/{8-9}] removing replica r63/2
I200401 06:20:59.380388 161120 kv/kvserver/store_remove_replica.go:122  [n3,s3,r35/3:/Table/53/1/{8-9}] removing replica r63/3
I200401 06:20:59.396313 159377 kv/kvserver/store_remove_replica.go:122  [n1,s1,r35/1:/Table/53/1/{8-9}] removing replica r63/1
I200401 06:20:59.520628 161062 kv/kvserver/store_remove_replica.go:122  [n3,s3,r39/3:/Table/53/1/{6-7}] removing replica r37/3
I200401 06:20:59.576951 160795 kv/kvserver/store_remove_replica.go:122  [n2,s2,r39/2:/Table/53/1/{6-7}] removing replica r37/2
I200401 06:20:59.589353 159365 kv/kvserver/store_remove_replica.go:122  [n1,s1,r39/1:/Table/53/1/{6-7}] removing replica r37/1
I200401 06:20:59.769139 159344 kv/kvserver/store_remove_replica.go:122  [n1,s1,r33/1:/Table/53{-/1/1}] removing replica r40/1
I200401 06:20:59.774730 160778 kv/kvserver/store_remove_replica.go:122  [n2,s2,r33/2:/Table/53{-/1/1}] removing replica r40/2
I200401 06:20:59.810462 161120 kv/kvserver/store_remove_replica.go:122  [n3,s3,r33/3:/Table/53{-/1/1}] removing replica r40/3
I200401 06:21:00.132867 164569 sql/table.go:942  [n1,client=127.0.0.1:37780,hostssl,user=root] queued new schema change job for table 53, mutation 0
I200401 06:21:00.133134 164569 sql/sqlbase/structured.go:1554  [n1,client=127.0.0.1:37780,hostssl,user=root] publish: descID=53 (bank) version=2 mtime=1970-01-01 00:00:00 +0000 UTC
I200401 06:21:00.755105 167273 kv/kvserver/replica_command.go:663  [n3,merge,s3,r53/3:/Table/53/1/{4-6}] initiating a merge of r39:/Table/53/1/{6-8} [(n1,s1):1, (n2,s2):2, (n3,s3):3, next=4, gen=10] into this range (lhs+rhs has (size=5.4 KiB+5.2 KiB=11 KiB qps=5.88+2.94=8.82qps) below threshold (size=128 MiB, qps=1250.00))
I200401 06:21:01.129702 160847 server/status/runtime.go:498  [n2] runtime stats: 1.3 GiB RSS, 794 goroutines, 61 MiB/108 MiB/118 MiB GO alloc/idle/total, 98 MiB/145 MiB CGO alloc/total, 1190.0 CGO/sec, 195.3/14.4 %(u/s)time, 0.8 %gc (11x), 4.0 MiB/4.0 MiB (r/w)net
I200401 06:21:01.136534 164569 sql/table.go:942  [n1,client=127.0.0.1:37780,hostssl,user=root] queued new schema change job for table 53, mutation 0
I200401 06:21:01.146874 164569 sql/sqlbase/structured.go:1554  [n1,client=127.0.0.1:37780,hostssl,user=root] publish: descID=53 (bank) version=2 mtime=1970-01-01 00:00:00 +0000 UTC
I200401 06:21:01.210635 159373 kv/kvserver/store_remove_replica.go:122  [n1,s1,r53/1:/Table/53/1/{4-6}] removing replica r39/1
I200401 06:21:01.222381 160786 kv/kvserver/store_remove_replica.go:122  [n2,s2,r53/2:/Table/53/1/{4-6}] removing replica r39/2
I200401 06:21:01.230503 161110 kv/kvserver/store_remove_replica.go:122  [n3,s3,r53/3:/Table/53/1/{4-6}] removing replica r39/3
I200401 06:21:01.842153 167415 kv/kvserver/replica_command.go:663  [n1,merge,s1,r33/1:/Table/53{-/1/4}] initiating a merge of r53:/Table/53/1/{4-8} [(n1,s1):1, (n2,s2):2, (n3,s3):3, next=4, gen=11] into this range (lhs+rhs has (size=12 KiB+11 KiB=23 KiB qps=34.65+17.03=51.68qps) below threshold (size=128 MiB, qps=1250.00))
I200401 06:21:01.894026 164569 sql/table.go:942  [n1,client=127.0.0.1:37780,hostssl,user=root] queued new schema change job for table 53, mutation 0
I200401 06:21:01.894400 164569 sql/sqlbase/structured.go:1554  [n1,client=127.0.0.1:37780,hostssl,user=root] publish: descID=53 (bank) version=2 mtime=1970-01-01 00:00:00 +0000 UTC
I200401 06:21:02.129602 161134 server/status/runtime.go:498  [n3] runtime stats: 1.3 GiB RSS, 796 goroutines, 63 MiB/106 MiB/120 MiB GO alloc/idle/total, 99 MiB/145 MiB CGO alloc/total, 1248.0 CGO/sec, 195.5/14.9 %(u/s)time, 0.9 %gc (11x), 4.0 MiB/4.0 MiB (r/w)net
I200401 06:21:02.784327 161109 kv/kvserver/store_remove_replica.go:122  [n3,s3,r33/3:/Table/53{-/1/4}] removing replica r53/3
I200401 06:21:02.789208 160805 kv/kvserver/store_remove_replica.go:122  [n2,s2,r33/2:/Table/53{-/1/4}] removing replica r53/2
I200401 06:21:02.790909 159338 kv/kvserver/store_remove_replica.go:122  [n1,s1,r33/1:/Table/53{-/1/4}] removing replica r53/1
I200401 06:21:02.929439 164569 sql/table.go:942  [n1,client=127.0.0.1:37780,hostssl,user=root] queued new schema change job for table 53, mutation 0
I200401 06:21:02.929654 164569 sql/sqlbase/structured.go:1554  [n1,client=127.0.0.1:37780,hostssl,user=root] publish: descID=53 (bank) version=2 mtime=1970-01-01 00:00:00 +0000 UTC
I200401 06:21:03.543818 164569 sql/event_log.go:132  [n1,client=127.0.0.1:37780,hostssl,user=root] Event: "drop_table", target: 53, info: {TableName:data.public.bank Statement:DROP TABLE data.public.bank User:root CascadeDroppedViews:[]}
I200401 06:21:03.544093 164569 sql/event_log.go:132  [n1,client=127.0.0.1:37780,hostssl,user=root] Event: "drop_table", target: 53, info: {TableName:data.public.bank Statement:DROP TABLE data.public.bank User:root CascadeDroppedViews:[]}
I200401 06:21:03.544196 164569 sql/event_log.go:132  [n1,client=127.0.0.1:37780,hostssl,user=root] Event: "drop_table", target: 53, info: {TableName:data.public.bank Statement:DROP TABLE data.public.bank User:root CascadeDroppedViews:[]}
I200401 06:21:03.544295 164569 sql/event_log.go:132  [n1,client=127.0.0.1:37780,hostssl,user=root] Event: "drop_table", target: 53, info: {TableName:data.public.bank Statement:DROP TABLE data.public.bank User:root CascadeDroppedViews:[]}
I200401 06:21:03.544408 164569 sql/event_log.go:132  [n1,client=127.0.0.1:37780,hostssl,user=root] Event: "drop_table", target: 53, info: {TableName:data.public.bank Statement:DROP TABLE data.public.bank User:root CascadeDroppedViews:[]}
I200401 06:21:03.544517 164569 sql/event_log.go:132  [n1,client=127.0.0.1:37780,hostssl,user=root] Event: "drop_table", target: 53, info: {TableName:data.public.bank Statement:DROP TABLE data.public.bank User:root CascadeDroppedViews:[]}
I200401 06:21:03.544611 164569 sql/event_log.go:132  [n1,client=127.0.0.1:37780,hostssl,user=root] Event: "drop_table", target: 53, info: {TableName:data.public.bank Statement:DROP TABLE data.public.bank User:root CascadeDroppedViews:[]}
I200401 06:21:03.544952 164569 jobs/registry.go:285  [n1,client=127.0.0.1:37780,hostssl,user=root] scheduled jobs [542807368717959169]
I200401 06:21:04.255746 167808 kv/kvserver/replica_command.go:663  [n1,merge,s1,r33/1:/Table/53{-/1/8}] initiating a merge of r35:/{Table/53/1/8-Max} [(n1,s1):1, (n2,s2):2, (n3,s3):3, next=4, gen=9] into this range (lhs+rhs has (size=26 KiB+8.9 KiB=35 KiB qps=38.73+3.26=41.99qps) below threshold (size=128 MiB, qps=1250.00))
I200401 06:21:04.606085 159481 jobs/jobs.go:743  job 542807368717959169: adopting
I200401 06:21:05.030022 159379 kv/kvserver/store_remove_replica.go:122  [n1,s1,r33/1:/Table/53{-/1/8}] removing replica r35/1
I200401 06:21:05.065092 161090 kv/kvserver/store_remove_replica.go:122  [n3,s3,r33/3:/Table/53{-/1/8}] removing replica r35/3
I200401 06:21:05.082791 160774 kv/kvserver/store_remove_replica.go:122  [n2,s2,r33/2:/Table/53{-/1/8}] removing replica r35/2
I200401 06:21:05.376140 159481 jobs/registry.go:1178  job 542807368717959169: resuming execution
I200401 06:21:05.445842 168136 jobs/registry.go:797  [n1] job 542807368717959169: stepping through state running with error <nil>
I200401 06:21:05.451277 168136 sql/schema_changer.go:445  [n1,job=542807368717959169,scExec] schema change on bank (53 v2) mutation 0 starting execution...
I200401 06:21:05.487214 168136 sql/sqlbase/structured.go:1554  [n1,job=542807368717959169,scExec] publish: descID=53 (bank) version=3 mtime=1970-01-01 00:00:00 +0000 UTC
E200401 06:21:05.731677 165145 sql/conn_executor.go:796  [n2,client=127.0.0.1:38792,hostssl,user=root] a SQL panic has occurred while executing "UPDATE data.bank SET payload = $1 WHERE id = $2": event RetriableErr{CanAutoRetry:false, IsCommit:false} inappropriate in current state NoTxn{}
E200401 06:21:05.732647 165145 util/log/crash_reporting.go:208  [n2,client=127.0.0.1:38792,hostssl,user=root] a panic has occurred!

```

</p></details>

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestBackupRestoreWithConcurrentWrites PKG=./pkg/ccl/backupccl TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```

Related:
- #46827 ccl/backupccl: TestBackupRestoreWithConcurrentWrites failed  [C-test-failure](https://api.github.com/repos/cockroachdb/cockroach/labels/C-test-failure) [O-robot](https://api.github.com/repos/cockroachdb/cockroach/labels/O-robot) [branch-release-20.1](https://api.github.com/repos/cockroachdb/cockroach/labels/branch-release-20.1)

[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestBackupRestoreWithConcurrentWrites.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(ccl/backupccl).TestBackupRestoreWithConcurrentWrites failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1843022&tab=buildLog) on [master@3a8fb889e980b174bc72bece314d7061aec78c04](https://github.com/cockroachdb/cockroach/commits/3a8fb889e980b174bc72bece314d7061aec78c04):

Fatal error:
```
panic: event RetriableErr{CanAutoRetry:false, IsCommit:false} inappropriate in current state NoTxn{} [recovered]
	panic: panic while executing 1 statements: UPDATE _._ SET _ = $1 WHERE _ = $2; caused by event RetriableErr{CanAutoRetry:false, IsCommit:false} inappropriate in current state NoTxn{}

```

Stack:
```
goroutine 169013 [running]:
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).closeWrapper(0xc002ea8000, 0x6eab420, 0xc0057ff240, 0x5c52da0, 0xc000858ec0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:810 +0x4e1
github.com/cockroachdb/cockroach/pkg/sql.(*Server).ServeConn.func1(0xc002ea8000, 0x6eab420, 0xc0057ff240)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:474 +0x6f
panic(0x5c52da0, 0xc000858ec0)
	/usr/local/go/src/runtime/panic.go:679 +0x1b2
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).txnStateTransitionsApplyWrapper(0xc002ea8000, 0x6e2dce0, 0xc000858ea0, 0x5c517c0, 0xc0017fda40, 0x7f589e4708e8, 0xc0020d0180, 0x62, 0x0, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:2085 +0x16b5
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execCmd(0xc002ea8000, 0x6eab420, 0xc0057ff240, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1497 +0xa77
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).run(0xc002ea8000, 0x6eab420, 0xc0057ff240, 0xc00337c110, 0x5400, 0x15000, 0xc00337c1b0, 0xc0004cf510, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1297 +0x484
github.com/cockroachdb/cockroach/pkg/sql.(*Server).ServeConn(0xc006ee9080, 0x6eab420, 0xc0057ff240, 0xc002ea8000, 0x5400, 0x15000, 0xc00337c1b0, 0xc0004cf510, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:476 +0x132
github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*conn).processCommandsAsync.func1(0xc0032cd022, 0xc00596fb80, 0x6eab420, 0xc0057ff240, 0xc0004cf510, 0xc006ee9080, 0xc0020cfb00, 0x6ecf760, 0xc003e0a840, 0xc0016b8840, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn.go:594 +0x36c
created by github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*conn).processCommandsAsync
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn.go:526 +0x1ea

```

<details><summary>Log preceding fatal error</summary><p>

```
I200401 09:59:17.509122 171468 kv/kvserver/replica_command.go:663  [n1,merge,s1,r36/1:/Table/53/1/{2-3}] initiating a merge of r35:/Table/53/1/{3-4} [(n1,s1):1, (n2,s2):2, (n3,s3):3, next=4, gen=8] into this range (lhs+rhs has (size=973 B+489 B=1.4 KiB qps=2.40+2.64=5.05qps) below threshold (size=128 MiB, qps=1250.00))
I200401 09:59:17.938325 164551 kv/kvserver/store_remove_replica.go:122  [n1,s1,r36/1:/Table/53/1/{2-3}] removing replica r35/1
I200401 09:59:17.950118 166302 kv/kvserver/store_remove_replica.go:122  [n3,s3,r36/3:/Table/53/1/{2-3}] removing replica r35/3
I200401 09:59:17.967535 166030 kv/kvserver/store_remove_replica.go:122  [n2,s2,r36/2:/Table/53/1/{2-3}] removing replica r35/2
I200401 09:59:17.987868 169966 sql/event_log.go:132  [n1,client=127.0.0.1:42596,hostssl,user=root] Event: "drop_table", target: 53, info: {TableName:data.public.bank Statement:DROP TABLE data.public.bank User:root CascadeDroppedViews:[]}
I200401 09:59:17.988510 169966 jobs/registry.go:285  [n1,client=127.0.0.1:42596,hostssl,user=root] scheduled jobs [542850276208410625]
I200401 09:59:18.252392 171559 kv/kvserver/replica_command.go:663  [n2,merge,s2,r45/2:/Table/53/1/{6-7}] initiating a merge of r44:/Table/53/1/{7-8} [(n1,s1):1, (n2,s2):2, (n3,s3):3, next=4, gen=8] into this range (lhs+rhs has (size=973 B+731 B=1.7 KiB qps=3.09+5.16=8.24qps) below threshold (size=128 MiB, qps=1250.00))
I200401 09:59:18.269299 171667 kv/kvserver/replica_command.go:663  [n1,merge,s1,r36/1:/Table/53/1/{2-4}] initiating a merge of r37:/Table/53/1/{4-5} [(n1,s1):1, (n2,s2):2, (n3,s3):3, next=4, gen=8] into this range (lhs+rhs has (size=1.4 KiB+852 B=2.3 KiB qps=2.40+2.64=5.05qps) below threshold (size=128 MiB, qps=1250.00))
I200401 09:59:18.754075 164746 jobs/jobs.go:743  job 542850276208410625: adopting
I200401 09:59:18.788276 171864 kv/kvserver/replica_command.go:663  [n3,merge,s3,r44/3:/Table/53/1/{7-8}] initiating a merge of r43:/Table/53/1/{8-9} [(n1,s1):1, (n2,s2):2, (n3,s3):3, next=4, gen=8] into this range (lhs+rhs has (size=731 B+852 B=1.5 KiB qps=5.16+2.76=7.92qps) below threshold (size=128 MiB, qps=1250.00))
I200401 09:59:18.907537 164553 kv/kvserver/store_remove_replica.go:122  [n1,s1,r36/1:/Table/53/1/{2-4}] removing replica r37/1
I200401 09:59:18.925888 166005 kv/kvserver/store_remove_replica.go:122  [n2,s2,r36/2:/Table/53/1/{2-4}] removing replica r37/2
I200401 09:59:18.938302 166309 kv/kvserver/store_remove_replica.go:122  [n3,s3,r36/3:/Table/53/1/{2-4}] removing replica r37/3
I200401 09:59:19.266797 171937 kv/kvserver/replica_command.go:663  [n1,merge,s1,r33/1:/Table/53{-/1/1}] initiating a merge of r38:/Table/53/1/{1-2} [(n1,s1):1, (n2,s2):2, (n3,s3):3, next=4, gen=9] into this range (lhs+rhs has (size=489 B+1.1 KiB=1.5 KiB qps=0.00+36.39=36.39qps) below threshold (size=128 MiB, qps=1250.00))
I200401 09:59:19.373442 164746 jobs/registry.go:1178  job 542850276208410625: resuming execution
I200401 09:59:19.658012 172113 jobs/registry.go:797  [n1] job 542850276208410625: stepping through state running with error <nil>
I200401 09:59:19.761021 172113 sql/schema_changer.go:445  [n1,job=542850276208410625,scExec] schema change on bank (53 v2) mutation 0 starting execution...
I200401 09:59:19.841616 164288 server/status/runtime.go:498  [n1] runtime stats: 1.3 GiB RSS, 809 goroutines, 101 MiB/66 MiB/158 MiB GO alloc/idle/total, 100 MiB/149 MiB CGO alloc/total, 1082.5 CGO/sec, 193.7/16.1 %(u/s)time, 0.5 %gc (7x), 3.8 MiB/3.8 MiB (r/w)net
I200401 09:59:19.903073 164552 kv/kvserver/store_remove_replica.go:122  [n1,s1,r33/1:/Table/53{-/1/1}] removing replica r38/1
I200401 09:59:19.918426 166027 kv/kvserver/store_remove_replica.go:122  [n2,s2,r33/2:/Table/53{-/1/1}] removing replica r38/2
I200401 09:59:19.949592 166317 kv/kvserver/store_remove_replica.go:122  [n3,s3,r33/3:/Table/53{-/1/1}] removing replica r38/3
I200401 09:59:20.071773 172113 sql/lease.go:337  [n1,job=542850276208410625,scExec] waiting for 2 leases to expire: desc=[{bank 53 1}]
I200401 09:59:20.076642 166292 kv/kvserver/store_remove_replica.go:122  [n3,s3,r44/3:/Table/53/1/{7-8}] removing replica r43/3
I200401 09:59:20.079537 165996 kv/kvserver/store_remove_replica.go:122  [n2,s2,r44/2:/Table/53/1/{7-8}] removing replica r43/2
I200401 09:59:20.123581 164521 kv/kvserver/store_remove_replica.go:122  [n1,s1,r44/1:/Table/53/1/{7-8}] removing replica r43/1
I200401 09:59:20.246200 171559 kv/kvserver/replica_command.go:663  [n2,merge,s2,r45/2:/Table/53/1/{6-7}] initiating a merge of r44:/Table/53/1/{7-9} [(n1,s1):1, (n2,s2):2, (n3,s3):3, next=4, gen=9] into this range (lhs+rhs has (size=973 B+731 B=1.7 KiB qps=3.09+5.16=8.24qps) below threshold (size=128 MiB, qps=1250.00))
I200401 09:59:20.972429 172113 sql/lease.go:337  [n1,job=542850276208410625,scExec] waiting for 1 leases to expire: desc=[{bank 53 1}]
I200401 09:59:21.034715 172321 kv/kvserver/replica_command.go:663  [n1,merge,s1,r33/1:/Table/53{-/1/2}] initiating a merge of r36:/Table/53/1/{2-5} [(n1,s1):1, (n2,s2):2, (n3,s3):3, next=4, gen=10] into this range (lhs+rhs has (size=1.5 KiB+2.3 KiB=3.8 KiB qps=7.78+3.06=10.85qps) below threshold (size=128 MiB, qps=1250.00))
I200401 09:59:21.236300 171559 kv/kvserver/replica_command.go:663  [n2,merge,s2,r45/2:/Table/53/1/{6-7}] initiating a merge of r44:/Table/53/1/{7-9} [(n1,s1):1, (n2,s2):2, (n3,s3):3, next=4, gen=9] into this range (lhs+rhs has (size=973 B+731 B=1.7 KiB qps=3.09+5.16=8.24qps) below threshold (size=128 MiB, qps=1250.00))
I200401 09:59:21.945336 165990 kv/kvserver/store_remove_replica.go:122  [n2,s2,r45/2:/Table/53/1/{6-7}] removing replica r44/2
I200401 09:59:21.956158 164522 kv/kvserver/store_remove_replica.go:122  [n1,s1,r45/1:/Table/53/1/{6-7}] removing replica r44/1
I200401 09:59:21.970639 166313 kv/kvserver/store_remove_replica.go:122  [n3,s3,r45/3:/Table/53/1/{6-7}] removing replica r44/3
I200401 09:59:22.159705 172461 kv/kvserver/replica_command.go:663  [n2,merge,s2,r34/2:/Table/53/1/{5-6}] initiating a merge of r45:/Table/53/1/{6-9} [(n1,s1):1, (n2,s2):2, (n3,s3):3, next=4, gen=10] into this range (lhs+rhs has (size=610 B+2.5 KiB=3.1 KiB qps=0.00+7.85=7.85qps) below threshold (size=128 MiB, qps=1250.00))
I200401 09:59:22.263929 166314 kv/kvserver/store_remove_replica.go:122  [n3,s3,r33/3:/Table/53{-/1/2}] removing replica r36/3
I200401 09:59:22.269407 164530 kv/kvserver/store_remove_replica.go:122  [n1,s1,r33/1:/Table/53{-/1/2}] removing replica r36/1
I200401 09:59:22.313354 165991 kv/kvserver/store_remove_replica.go:122  [n2,s2,r33/2:/Table/53{-/1/2}] removing replica r36/2
I200401 09:59:22.854005 165977 kv/kvserver/store_remove_replica.go:122  [n2,s2,r34/2:/Table/53/1/{5-6}] removing replica r45/2
I200401 09:59:22.868587 164568 kv/kvserver/store_remove_replica.go:122  [n1,s1,r34/1:/Table/53/1/{5-6}] removing replica r45/1
I200401 09:59:22.857109 166327 kv/kvserver/store_remove_replica.go:122  [n3,s3,r34/3:/Table/53/1/{5-6}] removing replica r45/3
I200401 09:59:22.898719 172584 kv/kvserver/replica_command.go:663  [n1,merge,s1,r33/1:/Table/53{-/1/5}] initiating a merge of r34:/Table/53/1/{5-9} [(n1,s1):1, (n2,s2):2, (n3,s3):3, next=4, gen=11] into this range (lhs+rhs has (size=3.8 KiB+610 B=4.4 KiB qps=16.16+0.00=16.16qps) below threshold (size=128 MiB, qps=1250.00))
I200401 09:59:23.288069 172634 kv/kvserver/replica_command.go:663  [n2,merge,s2,r34/2:/Table/53/1/{5-9}] initiating a merge of r39:/{Table/53/1/9-Max} [(n1,s1):1, (n2,s2):2, (n3,s3):3, next=4, gen=8] into this range (lhs+rhs has (size=3.1 KiB+1.1 KiB=4.2 KiB qps=6.81+0.00=6.81qps) below threshold (size=128 MiB, qps=1250.00))
I200401 09:59:23.816657 172113 sql/sqlbase/structured.go:1554  [n1,job=542850276208410625,scExec] publish: descID=53 (bank) version=3 mtime=1970-01-01 00:00:00 +0000 UTC
I200401 09:59:23.889660 166069 server/status/runtime.go:498  [n2] runtime stats: 1.3 GiB RSS, 804 goroutines, 70 MiB/96 MiB/159 MiB GO alloc/idle/total, 100 MiB/149 MiB CGO alloc/total, 1174.5 CGO/sec, 187.8/17.0 %(u/s)time, 0.7 %gc (7x), 4.1 MiB/4.1 MiB (r/w)net
I200401 09:59:24.055403 165990 kv/kvserver/store_remove_replica.go:122  [n2,s2,r34/2:/Table/53/1/{5-9}] removing replica r39/2
I200401 09:59:24.058990 166331 kv/kvserver/store_remove_replica.go:122  [n3,s3,r34/3:/Table/53/1/{5-9}] removing replica r39/3
I200401 09:59:24.086538 164522 kv/kvserver/store_remove_replica.go:122  [n1,s1,r34/1:/Table/53/1/{5-9}] removing replica r39/1
I200401 09:59:24.369069 172584 kv/kvserver/replica_command.go:663  [n1,merge,s1,r33/1:/Table/53{-/1/5}] initiating a merge of r34:/{Table/53/1/5-Max} [(n1,s1):1, (n2,s2):2, (n3,s3):3, next=4, gen=12] into this range (lhs+rhs has (size=3.8 KiB+610 B=4.4 KiB qps=16.16+0.00=16.16qps) below threshold (size=128 MiB, qps=1250.00))
I200401 09:59:24.745958 166371 server/status/runtime.go:498  [n3] runtime stats: 1.3 GiB RSS, 835 goroutines, 102 MiB/65 MiB/159 MiB GO alloc/idle/total, 101 MiB/151 MiB CGO alloc/total, 1193.0 CGO/sec, 187.6/17.5 %(u/s)time, 0.7 %gc (6x), 4.1 MiB/4.1 MiB (r/w)net
E200401 09:59:24.770868 169013 sql/conn_executor.go:796  [n1,client=127.0.0.1:42588,hostssl,user=root] a SQL panic has occurred while executing "UPDATE data.bank SET payload = $1 WHERE id = $2": event RetriableErr{CanAutoRetry:false, IsCommit:false} inappropriate in current state NoTxn{}
E200401 09:59:24.771943 169013 util/log/crash_reporting.go:208  [n1,client=127.0.0.1:42588,hostssl,user=root] a panic has occurred!

```

</p></details>

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestBackupRestoreWithConcurrentWrites PKG=./pkg/ccl/backupccl TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```

Related:
- #46827 ccl/backupccl: TestBackupRestoreWithConcurrentWrites failed  [C-test-failure](https://api.github.com/repos/cockroachdb/cockroach/labels/C-test-failure) [O-robot](https://api.github.com/repos/cockroachdb/cockroach/labels/O-robot) [branch-release-20.1](https://api.github.com/repos/cockroachdb/cockroach/labels/branch-release-20.1)

[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestBackupRestoreWithConcurrentWrites.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

--- a/pkg/sql/conn_executor_prepare.go
+++ b/pkg/sql/conn_executor_prepare.go
@@ -164,34 +164,32 @@ func (ex *connExecutor) prepare(
 	// Preparing needs a transaction because it needs to retrieve db/table
 	// descriptors for type checking. If we already have an open transaction for
 	// this planner, use it. Using the user's transaction here is critical for
-	// proper deadlock detection. At the time of writing it is the case that any
+	// proper deadlock detection. At the time of writing, it is the case that any
 	// data read on behalf of this transaction is not cached for use in other
 	// transactions. It's critical that this fact remain true but nothing really
 	// enforces it. If we create a new transaction (newTxn is true), we'll need to
 	// finish it before we return.
-	newTxn, txn := false, ex.state.mu.txn
-	if txn == nil || !txn.IsOpen() {
-		newTxn, txn = true, kv.NewTxn(ctx, ex.server.cfg.DB, ex.server.cfg.NodeID.Get())
+
+	var flags planFlags
+	prepare := func(ctx context.Context, txn *kv.Txn) (err error) {
+		ex.statsCollector.reset(&ex.server.sqlStats, ex.appStats, &ex.phaseTimes)
+		p := &ex.planner
+		ex.resetPlanner(ctx, p, txn, ex.server.cfg.Clock.PhysicalTime() /* stmtTS */)
+		p.stmt = &stmt
+		p.semaCtx.Annotations = tree.MakeAnnotations(stmt.NumAnnotations)
+		flags, err = ex.populatePrepared(ctx, txn, placeholderHints, p)
+		return err
 	}
 
-	ex.statsCollector.reset(&ex.server.sqlStats, ex.appStats, &ex.phaseTimes)
-	p := &ex.planner
-	ex.resetPlanner(ctx, p, txn, ex.server.cfg.Clock.PhysicalTime() /* stmtTS */)
-	p.stmt = &stmt
-	p.semaCtx.Annotations = tree.MakeAnnotations(stmt.NumAnnotations)
-	flags, err := ex.populatePrepared(ctx, txn, placeholderHints, p)
-	if err != nil {
-		// NB: if this is not a new transaction then let the connExecutor state
-		// machine decide whether we should clean up intents; we may be restarting
-		// and want to leave them in place.
-		if newTxn {
-			txn.CleanupOnError(ctx, err)
+	if txn := ex.state.mu.txn; txn != nil && txn.IsOpen() {
+		// Use the existing transaction.
+		if err := prepare(ctx, txn); err != nil {
+			return nil, err
 		}
-		return nil, err
-	}
-	if newTxn {
-		// Clean up the newly created transaction if we made one.
-		if err := txn.CommitOrCleanup(ctx); err != nil {
+	} else {
+		// Use a new transaction. This will handle retriable errors here rather
+		// than bubbling them up to the connExecutor state machine.
+		if err := ex.server.cfg.DB.Txn(ctx, prepare); err != nil {
 			return nil, err
 		}
 	}
@@ -209,6 +207,11 @@ func (ex *connExecutor) prepare(
 func (ex *connExecutor) populatePrepared(
 	ctx context.Context, txn *kv.Txn, placeholderHints tree.PlaceholderTypes, p *planner,
 ) (planFlags, error) {
+	if before := ex.server.cfg.TestingKnobs.BeforePrepare; before != nil {
+		if err := before(ctx, ex.planner.stmt.String(), txn); err != nil {
+			return 0, err
+		}
+	}
 	stmt := p.stmt
 	if err := p.semaCtx.Placeholders.Init(stmt.NumPlaceholders, placeholderHints); err != nil {
 		return 0, err

--- a/pkg/sql/conn_executor_test.go
+++ b/pkg/sql/conn_executor_test.go
@@ -23,6 +23,7 @@ import (
 
 	"github.com/cockroachdb/cockroach/pkg/base"
 	"github.com/cockroachdb/cockroach/pkg/keys"
+	"github.com/cockroachdb/cockroach/pkg/kv"
 	"github.com/cockroachdb/cockroach/pkg/kv/kvserver"
 	"github.com/cockroachdb/cockroach/pkg/kv/kvserver/storagebase"
 	"github.com/cockroachdb/cockroach/pkg/roachpb"
@@ -629,6 +630,37 @@ func TestPrepareInExplicitTransactionDoesNotDeadlock(t *testing.T) {
 			require.Regexp(t, "RETRY_SERIALIZABLE", tx2CommitErr)
 		}
 	}
+}
+
+// TestRetriableErrorDuringPrepare ensures that when preparing and using a new
+// transaction, retriable errors are handled properly and do not propagate to
+// the user's transaction.
+func TestRetriableErrorDuringPrepare(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	const uniqueString = "'a very unique string'"
+	var failed int64
+	const numToFail = 2 // only fail on the first two attempts
+	s, sqlDB, _ := serverutils.StartServer(t, base.TestServerArgs{
+		Knobs: base.TestingKnobs{
+			SQLExecutor: &sql.ExecutorTestingKnobs{
+				BeforePrepare: func(ctx context.Context, stmt string, txn *kv.Txn) error {
+					if strings.Contains(stmt, uniqueString) && atomic.AddInt64(&failed, 1) <= numToFail {
+						return roachpb.NewTransactionRetryWithProtoRefreshError("boom",
+							txn.ID(), *txn.TestingCloneTxn())
+					}
+					return nil
+				},
+			},
+		},
+	})
+	defer s.Stopper().Stop(context.Background())
+
+	testDB := sqlutils.MakeSQLRunner(sqlDB)
+	testDB.Exec(t, "CREATE TABLE foo (i INT PRIMARY KEY)")
+
+	stmt, err := sqlDB.Prepare("SELECT " + uniqueString)
+	require.NoError(t, err)
+	defer func() { _ = stmt.Close() }()
 }
 
 // This test ensures that when in an explicit transaction and statement

--- a/pkg/sql/exec_util.go
+++ b/pkg/sql/exec_util.go
@@ -652,6 +652,10 @@ type ExecutorTestingKnobs struct {
 	// statement has been executed.
 	StatementFilter StatementFilter
 
+	// BeforePrepare can be used to trap execution of SQL statement preparation.
+	// If a nil error is returned, planning continues as usual.
+	BeforePrepare func(ctx context.Context, stmt string, txn *kv.Txn) error
+
 	// BeforeExecute is called by the Executor before plan execution. It is useful
 	// for synchronizing statement execution.
 	BeforeExecute func(ctx context.Context, stmt string)

